<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Appendix B: analyzing scRNA-seq data by existing method II | Introduction to ASURAT (version 0.0.0.9001)</title>
  <meta name="description" content="This is a developmental version of ASURAT." />
  <meta name="generator" content="bookdown 0.23 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Appendix B: analyzing scRNA-seq data by existing method II | Introduction to ASURAT (version 0.0.0.9001)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a developmental version of ASURAT." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Appendix B: analyzing scRNA-seq data by existing method II | Introduction to ASURAT (version 0.0.0.9001)" />
  
  <meta name="twitter:description" content="This is a developmental version of ASURAT." />
  

<meta name="author" content="Keita Iida" />


<meta name="date" content="2021-09-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"/>
<link rel="next" href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#license-and-description"><i class="fa fa-check"></i><b>1.1</b> License and description</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computational-environment"><i class="fa fa-check"></i><b>1.2</b> Computational environment</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dataset.html"><a href="dataset.html"><i class="fa fa-check"></i><b>2</b> Preparing data sets</a>
<ul>
<li class="chapter" data-level="2.1" data-path="dataset.html"><a href="dataset.html#case-i-10x-data-stewart-et-al.-nat-cancer-2020"><i class="fa fa-check"></i><b>2.1</b> Case I: 10x data (Stewart et al., Nat Cancer, 2020)</a></li>
<li class="chapter" data-level="2.2" data-path="dataset.html"><a href="dataset.html#case-ii-pdac-a-st-and-indrop-moncada-et-al.-nat-biotechnol-2020"><i class="fa fa-check"></i><b>2.2</b> Case II: PDAC-A ST and inDrop (Moncada et al., Nat Biotechnol, 2020)</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="dataset.html"><a href="dataset.html#pdac-a-st"><i class="fa fa-check"></i><b>2.2.1</b> PDAC-A ST</a></li>
<li class="chapter" data-level="2.2.2" data-path="dataset.html"><a href="dataset.html#pdac-a-indrop"><i class="fa fa-check"></i><b>2.2.2</b> PDAC-A inDrop</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="dataset.html"><a href="dataset.html#case-iii-input-10x-pbmc-data"><i class="fa fa-check"></i><b>2.3</b> Case III: input 10x PBMC data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataqc.html"><a href="dataqc.html"><i class="fa fa-check"></i><b>3</b> Data quality control (QC)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="dataqc.html"><a href="dataqc.html#quick-qc"><i class="fa fa-check"></i><b>3.1</b> Quick QC</a></li>
<li class="chapter" data-level="3.2" data-path="dataqc.html"><a href="dataqc.html#sample-qc"><i class="fa fa-check"></i><b>3.2</b> Sample QC</a></li>
<li class="chapter" data-level="3.3" data-path="dataqc.html"><a href="dataqc.html#variable-qc"><i class="fa fa-check"></i><b>3.3</b> Variable QC</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>4</b> Normalizing and centering data</a></li>
<li class="chapter" data-level="5" data-path="integration.html"><a href="integration.html"><i class="fa fa-check"></i><b>5</b> Data integration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="integration.html"><a href="integration.html#integrate-pdac-indrop-and-st-data-by-seurat"><i class="fa fa-check"></i><b>5.1</b> Integrate PDAC inDrop and ST data by Seurat</a></li>
<li class="chapter" data-level="5.2" data-path="integration.html"><a href="integration.html#identifying-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.2</b> Identifying differentially expressed genes</a></li>
<li class="chapter" data-level="5.3" data-path="integration.html"><a href="integration.html#inferring-cell-cycle-phases"><i class="fa fa-check"></i><b>5.3</b> Inferring cell cycle phases</a></li>
<li class="chapter" data-level="5.4" data-path="integration.html"><a href="integration.html#save-results"><i class="fa fa-check"></i><b>5.4</b> Save results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="computcorr.html"><a href="computcorr.html"><i class="fa fa-check"></i><b>6</b> Computing correlations among genes</a></li>
<li class="chapter" data-level="7" data-path="markergenes.html"><a href="markergenes.html"><i class="fa fa-check"></i><b>7</b> Checking expression profiles of marker genes</a></li>
<li class="chapter" data-level="8" data-path="collectdb.html"><a href="collectdb.html"><i class="fa fa-check"></i><b>8</b> Collecting databases (optional)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="collectdb.html"><a href="collectdb.html#disease-ontology-db"><i class="fa fa-check"></i><b>8.1</b> Disease Ontology DB</a></li>
<li class="chapter" data-level="8.2" data-path="collectdb.html"><a href="collectdb.html#cell-ontology-db"><i class="fa fa-check"></i><b>8.2</b> Cell Ontology DB</a></li>
<li class="chapter" data-level="8.3" data-path="collectdb.html"><a href="collectdb.html#gene-ontology-db"><i class="fa fa-check"></i><b>8.3</b> Gene Ontology DB</a></li>
<li class="chapter" data-level="8.4" data-path="collectdb.html"><a href="collectdb.html#kegg"><i class="fa fa-check"></i><b>8.4</b> KEGG</a></li>
<li class="chapter" data-level="8.5" data-path="collectdb.html"><a href="collectdb.html#custom-built-db-for-our-analysis-of-pdac-data"><i class="fa fa-check"></i><b>8.5</b> Custom-built DB for our analysis of PDAC data</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="cell_type.html"><a href="cell_type.html"><i class="fa fa-check"></i><b>9</b> Inferring cell types or diseases (optional)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="cell_type.html"><a href="cell_type.html#add-database-to-asurat-objects"><i class="fa fa-check"></i><b>9.1</b> Add database to ASURAT objects</a></li>
<li class="chapter" data-level="9.2" data-path="cell_type.html"><a href="cell_type.html#quick-qc-1"><i class="fa fa-check"></i><b>9.2</b> Quick QC</a></li>
<li class="chapter" data-level="9.3" data-path="cell_type.html"><a href="cell_type.html#correlation-graph-based-decomposition"><i class="fa fa-check"></i><b>9.3</b> Correlation graph-based decomposition</a></li>
<li class="chapter" data-level="9.4" data-path="cell_type.html"><a href="cell_type.html#creating-systems-of-signs"><i class="fa fa-check"></i><b>9.4</b> Creating systems of signs</a></li>
<li class="chapter" data-level="9.5" data-path="cell_type.html"><a href="cell_type.html#removing-redundant-signs"><i class="fa fa-check"></i><b>9.5</b> Removing redundant signs</a></li>
<li class="chapter" data-level="9.6" data-path="cell_type.html"><a href="cell_type.html#creating-sign-by-sample-matrices"><i class="fa fa-check"></i><b>9.6</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="9.7" data-path="cell_type.html"><a href="cell_type.html#dimensionality-reduction"><i class="fa fa-check"></i><b>9.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="cell_type.html"><a href="cell_type.html#principal-component-analysis-optional"><i class="fa fa-check"></i><b>9.7.1</b> Principal component analysis (optional)</a></li>
<li class="chapter" data-level="9.7.2" data-path="cell_type.html"><a href="cell_type.html#t-distributed-stochastic-neighbor-embedding-optional"><i class="fa fa-check"></i><b>9.7.2</b> t-distributed stochastic neighbor embedding (optional)</a></li>
<li class="chapter" data-level="9.7.3" data-path="cell_type.html"><a href="cell_type.html#uniform-manifold-approximation-and-projection-optional"><i class="fa fa-check"></i><b>9.7.3</b> Uniform manifold approximation and projection (optional)</a></li>
<li class="chapter" data-level="9.7.4" data-path="cell_type.html"><a href="cell_type.html#diffusion-map-optional"><i class="fa fa-check"></i><b>9.7.4</b> Diffusion map (optional)</a></li>
<li class="chapter" data-level="9.7.5" data-path="cell_type.html"><a href="cell_type.html#visualization"><i class="fa fa-check"></i><b>9.7.5</b> Visualization</a></li>
<li class="chapter" data-level="9.7.6" data-path="cell_type.html"><a href="cell_type.html#save-results."><i class="fa fa-check"></i><b>9.7.6</b> Save results.</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="cell_type.html"><a href="cell_type.html#cell-clustering"><i class="fa fa-check"></i><b>9.8</b> Cell clustering</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="cell_type.html"><a href="cell_type.html#partitioning-around-medoids-pam-optional"><i class="fa fa-check"></i><b>9.8.1</b> Partitioning around medoids (PAM) (optional)</a></li>
<li class="chapter" data-level="9.8.2" data-path="cell_type.html"><a href="cell_type.html#hierarchical-clustering-and-cutting-tree-optional"><i class="fa fa-check"></i><b>9.8.2</b> Hierarchical clustering and cutting tree (optional)</a></li>
<li class="chapter" data-level="9.8.3" data-path="cell_type.html"><a href="cell_type.html#community-detection-algorithm-with-k-nearest-neighbor-knn-graph-optional"><i class="fa fa-check"></i><b>9.8.3</b> Community detection algorithm with k-nearest neighbor (KNN) graph (optional)</a></li>
<li class="chapter" data-level="9.8.4" data-path="cell_type.html"><a href="cell_type.html#clustering-by-branch-detection-along-pseudotime-optional"><i class="fa fa-check"></i><b>9.8.4</b> Clustering by branch detection along pseudotime (optional)</a></li>
<li class="chapter" data-level="9.8.5" data-path="cell_type.html"><a href="cell_type.html#save-data"><i class="fa fa-check"></i><b>9.8.5</b> Save data</a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="cell_type.html"><a href="cell_type.html#visualize-clustering-results"><i class="fa fa-check"></i><b>9.9</b> Visualize clustering results</a></li>
<li class="chapter" data-level="9.10" data-path="cell_type.html"><a href="cell_type.html#find-significant-signs"><i class="fa fa-check"></i><b>9.10</b> Find significant signs</a>
<ul>
<li class="chapter" data-level="9.10.1" data-path="cell_type.html"><a href="cell_type.html#compute-separation-indices"><i class="fa fa-check"></i><b>9.10.1</b> Compute separation indices</a></li>
<li class="chapter" data-level="9.10.2" data-path="cell_type.html"><a href="cell_type.html#investigate-sign-scores"><i class="fa fa-check"></i><b>9.10.2</b> Investigate sign scores</a></li>
</ul></li>
<li class="chapter" data-level="9.11" data-path="cell_type.html"><a href="cell_type.html#investigate-batch-effects"><i class="fa fa-check"></i><b>9.11</b> Investigate batch effects</a></li>
<li class="chapter" data-level="9.12" data-path="cell_type.html"><a href="cell_type.html#spatial-mapping-by-spaniel"><i class="fa fa-check"></i><b>9.12</b> Spatial mapping by Spaniel</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biological_function.html"><a href="biological_function.html"><i class="fa fa-check"></i><b>10</b> Inferring biological functions (optional)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biological_function.html"><a href="biological_function.html#add-database-to-asurat-objects-1"><i class="fa fa-check"></i><b>10.1</b> Add database to ASURAT objects</a></li>
<li class="chapter" data-level="10.2" data-path="biological_function.html"><a href="biological_function.html#quick-qc-2"><i class="fa fa-check"></i><b>10.2</b> Quick QC</a></li>
<li class="chapter" data-level="10.3" data-path="biological_function.html"><a href="biological_function.html#correlation-graph-based-decomposition-1"><i class="fa fa-check"></i><b>10.3</b> Correlation graph-based decomposition</a></li>
<li class="chapter" data-level="10.4" data-path="biological_function.html"><a href="biological_function.html#creating-systems-of-signs-1"><i class="fa fa-check"></i><b>10.4</b> Creating systems of signs</a></li>
<li class="chapter" data-level="10.5" data-path="biological_function.html"><a href="biological_function.html#removing-redundant-signs-1"><i class="fa fa-check"></i><b>10.5</b> Removing redundant signs</a></li>
<li class="chapter" data-level="10.6" data-path="biological_function.html"><a href="biological_function.html#creating-sign-by-sample-matrices-1"><i class="fa fa-check"></i><b>10.6</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="10.7" data-path="biological_function.html"><a href="biological_function.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>10.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="biological_function.html"><a href="biological_function.html#principal-component-analysis-optional-1"><i class="fa fa-check"></i><b>10.7.1</b> Principal component analysis (optional)</a></li>
<li class="chapter" data-level="10.7.2" data-path="biological_function.html"><a href="biological_function.html#t-distributed-stochastic-neighbor-embedding-optional-1"><i class="fa fa-check"></i><b>10.7.2</b> t-distributed stochastic neighbor embedding (optional)</a></li>
<li class="chapter" data-level="10.7.3" data-path="biological_function.html"><a href="biological_function.html#uniform-manifold-approximation-and-projection-optional-1"><i class="fa fa-check"></i><b>10.7.3</b> Uniform manifold approximation and projection (optional)</a></li>
<li class="chapter" data-level="10.7.4" data-path="biological_function.html"><a href="biological_function.html#diffusion-map-optional-1"><i class="fa fa-check"></i><b>10.7.4</b> Diffusion map (optional)</a></li>
<li class="chapter" data-level="10.7.5" data-path="biological_function.html"><a href="biological_function.html#visualization-1"><i class="fa fa-check"></i><b>10.7.5</b> Visualization</a></li>
<li class="chapter" data-level="10.7.6" data-path="biological_function.html"><a href="biological_function.html#save-results.-1"><i class="fa fa-check"></i><b>10.7.6</b> Save results.</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="biological_function.html"><a href="biological_function.html#cell-clustering-1"><i class="fa fa-check"></i><b>10.8</b> Cell clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="biological_function.html"><a href="biological_function.html#partitioning-around-medoids-pam-optional-1"><i class="fa fa-check"></i><b>10.8.1</b> Partitioning around medoids (PAM) (optional)</a></li>
<li class="chapter" data-level="10.8.2" data-path="biological_function.html"><a href="biological_function.html#hierarchical-clustering-and-cutting-tree-optional-1"><i class="fa fa-check"></i><b>10.8.2</b> Hierarchical clustering and cutting tree (optional)</a></li>
<li class="chapter" data-level="10.8.3" data-path="biological_function.html"><a href="biological_function.html#community-detection-algorithm-with-k-nearest-neighbor-knn-graph-optional-1"><i class="fa fa-check"></i><b>10.8.3</b> Community detection algorithm with k-nearest neighbor (KNN) graph (optional)</a></li>
<li class="chapter" data-level="10.8.4" data-path="biological_function.html"><a href="biological_function.html#clustering-by-branch-detection-along-pseudotime-optional-1"><i class="fa fa-check"></i><b>10.8.4</b> Clustering by branch detection along pseudotime (optional)</a></li>
<li class="chapter" data-level="10.8.5" data-path="biological_function.html"><a href="biological_function.html#save-data-1"><i class="fa fa-check"></i><b>10.8.5</b> Save data</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="biological_function.html"><a href="biological_function.html#visualize-clustering-results-1"><i class="fa fa-check"></i><b>10.9</b> Visualize clustering results</a></li>
<li class="chapter" data-level="10.10" data-path="biological_function.html"><a href="biological_function.html#find-significant-signs-1"><i class="fa fa-check"></i><b>10.10</b> Find significant signs</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="biological_function.html"><a href="biological_function.html#compute-separation-indices-1"><i class="fa fa-check"></i><b>10.10.1</b> Compute separation indices</a></li>
<li class="chapter" data-level="10.10.2" data-path="biological_function.html"><a href="biological_function.html#investigate-sign-scores-1"><i class="fa fa-check"></i><b>10.10.2</b> Investigate sign scores</a></li>
</ul></li>
<li class="chapter" data-level="10.11" data-path="biological_function.html"><a href="biological_function.html#investigate-batch-effects-1"><i class="fa fa-check"></i><b>10.11</b> Investigate batch effects</a></li>
<li class="chapter" data-level="10.12" data-path="biological_function.html"><a href="biological_function.html#spatial-mapping-by-spaniel-1"><i class="fa fa-check"></i><b>10.12</b> Spatial mapping by Spaniel</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pathway.html"><a href="pathway.html"><i class="fa fa-check"></i><b>11</b> Inferring pathway activities (optional)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="pathway.html"><a href="pathway.html#add-database-to-asurat-objects-2"><i class="fa fa-check"></i><b>11.1</b> Add database to ASURAT objects</a></li>
<li class="chapter" data-level="11.2" data-path="pathway.html"><a href="pathway.html#quick-qc-3"><i class="fa fa-check"></i><b>11.2</b> Quick QC</a></li>
<li class="chapter" data-level="11.3" data-path="pathway.html"><a href="pathway.html#correlation-graph-based-decomposition-2"><i class="fa fa-check"></i><b>11.3</b> Correlation graph-based decomposition</a></li>
<li class="chapter" data-level="11.4" data-path="pathway.html"><a href="pathway.html#creating-systems-of-signs-2"><i class="fa fa-check"></i><b>11.4</b> Creating systems of signs</a></li>
<li class="chapter" data-level="11.5" data-path="pathway.html"><a href="pathway.html#removing-redundant-signs-2"><i class="fa fa-check"></i><b>11.5</b> Removing redundant signs</a></li>
<li class="chapter" data-level="11.6" data-path="pathway.html"><a href="pathway.html#creating-sign-by-sample-matrices-2"><i class="fa fa-check"></i><b>11.6</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="11.7" data-path="pathway.html"><a href="pathway.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>11.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="pathway.html"><a href="pathway.html#principal-component-analysis-optional-2"><i class="fa fa-check"></i><b>11.7.1</b> Principal component analysis (optional)</a></li>
<li class="chapter" data-level="11.7.2" data-path="pathway.html"><a href="pathway.html#t-distributed-stochastic-neighbor-embedding-optional-2"><i class="fa fa-check"></i><b>11.7.2</b> t-distributed stochastic neighbor embedding (optional)</a></li>
<li class="chapter" data-level="11.7.3" data-path="pathway.html"><a href="pathway.html#uniform-manifold-approximation-and-projection-optional-2"><i class="fa fa-check"></i><b>11.7.3</b> Uniform manifold approximation and projection (optional)</a></li>
<li class="chapter" data-level="11.7.4" data-path="pathway.html"><a href="pathway.html#diffusion-map-optional-2"><i class="fa fa-check"></i><b>11.7.4</b> Diffusion map (optional)</a></li>
<li class="chapter" data-level="11.7.5" data-path="pathway.html"><a href="pathway.html#visualization-2"><i class="fa fa-check"></i><b>11.7.5</b> Visualization</a></li>
<li class="chapter" data-level="11.7.6" data-path="pathway.html"><a href="pathway.html#save-results.-2"><i class="fa fa-check"></i><b>11.7.6</b> Save results.</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="pathway.html"><a href="pathway.html#cell-clustering-2"><i class="fa fa-check"></i><b>11.8</b> Cell clustering</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="pathway.html"><a href="pathway.html#partitioning-around-medoids-pam-optional-2"><i class="fa fa-check"></i><b>11.8.1</b> Partitioning around medoids (PAM) (optional)</a></li>
<li class="chapter" data-level="11.8.2" data-path="pathway.html"><a href="pathway.html#hierarchical-clustering-and-cutting-tree-optional-2"><i class="fa fa-check"></i><b>11.8.2</b> Hierarchical clustering and cutting tree (optional)</a></li>
<li class="chapter" data-level="11.8.3" data-path="pathway.html"><a href="pathway.html#community-detection-algorithm-with-k-nearest-neighbor-knn-graph-optional-2"><i class="fa fa-check"></i><b>11.8.3</b> Community detection algorithm with k-nearest neighbor (KNN) graph (optional)</a></li>
<li class="chapter" data-level="11.8.4" data-path="pathway.html"><a href="pathway.html#clustering-by-branch-detection-along-pseudotime-optional-2"><i class="fa fa-check"></i><b>11.8.4</b> Clustering by branch detection along pseudotime (optional)</a></li>
<li class="chapter" data-level="11.8.5" data-path="pathway.html"><a href="pathway.html#save-data-2"><i class="fa fa-check"></i><b>11.8.5</b> Save data</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="pathway.html"><a href="pathway.html#visualize-clustering-results-2"><i class="fa fa-check"></i><b>11.9</b> Visualize clustering results</a></li>
<li class="chapter" data-level="11.10" data-path="pathway.html"><a href="pathway.html#find-significant-signs-2"><i class="fa fa-check"></i><b>11.10</b> Find significant signs</a>
<ul>
<li class="chapter" data-level="11.10.1" data-path="pathway.html"><a href="pathway.html#compute-separation-indices-2"><i class="fa fa-check"></i><b>11.10.1</b> Compute separation indices</a></li>
<li class="chapter" data-level="11.10.2" data-path="pathway.html"><a href="pathway.html#investigate-sign-scores-2"><i class="fa fa-check"></i><b>11.10.2</b> Investigate sign scores</a></li>
</ul></li>
<li class="chapter" data-level="11.11" data-path="pathway.html"><a href="pathway.html#investigate-batch-effects-2"><i class="fa fa-check"></i><b>11.11</b> Investigate batch effects</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="multi_layered.html"><a href="multi_layered.html"><i class="fa fa-check"></i><b>12</b> Multi-layered analysis across all the signs</a>
<ul>
<li class="chapter" data-level="12.1" data-path="multi_layered.html"><a href="multi_layered.html#concatenate-asurat-objects"><i class="fa fa-check"></i><b>12.1</b> Concatenate ASURAT objects</a></li>
<li class="chapter" data-level="12.2" data-path="multi_layered.html"><a href="multi_layered.html#find-significant-signs-and-genes-across-different-data-types"><i class="fa fa-check"></i><b>12.2</b> Find significant signs and genes across different data types</a></li>
<li class="chapter" data-level="12.3" data-path="multi_layered.html"><a href="multi_layered.html#spatial-mapping-by-spaniel-2"><i class="fa fa-check"></i><b>12.3</b> Spatial mapping by Spaniel</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><i class="fa fa-check"></i><b>13</b> Appendix A: analyzing scRNA-seq data by existing method I</a>
<ul>
<li class="chapter" data-level="13.1" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#preprocessing"><i class="fa fa-check"></i><b>13.1</b> Preprocessing</a></li>
<li class="chapter" data-level="13.2" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#cluster-cells"><i class="fa fa-check"></i><b>13.2</b> Cluster cells</a></li>
<li class="chapter" data-level="13.3" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#visualize-data-in-low-dimensional-space"><i class="fa fa-check"></i><b>13.3</b> Visualize data in low dimensional space</a></li>
<li class="chapter" data-level="13.4" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#find-differentially-expressed-genes"><i class="fa fa-check"></i><b>13.4</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="13.5" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#save-results-1"><i class="fa fa-check"></i><b>13.5</b> Save results</a></li>
<li class="chapter" data-level="13.6" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#infer-cell-types-for-pbmc-datasets"><i class="fa fa-check"></i><b>13.6</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><i class="fa fa-check"></i><b>14</b> Appendix B: analyzing scRNA-seq data by existing method II</a>
<ul>
<li class="chapter" data-level="14.1" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#preprocessing-1"><i class="fa fa-check"></i><b>14.1</b> Preprocessing</a></li>
<li class="chapter" data-level="14.2" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cluster-cells-1"><i class="fa fa-check"></i><b>14.2</b> Cluster cells</a></li>
<li class="chapter" data-level="14.3" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#visualize-data-in-low-dimensional-space-1"><i class="fa fa-check"></i><b>14.3</b> Visualize data in low dimensional space</a></li>
<li class="chapter" data-level="14.4" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cell-cycle-scoring"><i class="fa fa-check"></i><b>14.4</b> Cell cycle scoring</a></li>
<li class="chapter" data-level="14.5" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#find-differentially-expressed-genes-1"><i class="fa fa-check"></i><b>14.5</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="14.6" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#enrichment-analysis"><i class="fa fa-check"></i><b>14.6</b> Enrichment analysis</a></li>
<li class="chapter" data-level="14.7" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#save-results-2"><i class="fa fa-check"></i><b>14.7</b> Save results</a></li>
<li class="chapter" data-level="14.8" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#infer-cell-types-for-pbmc-datasets-1"><i class="fa fa-check"></i><b>14.8</b> Infer cell types for PBMC datasets</a></li>
<li class="chapter" data-level="14.9" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#infer-cell-types-for-pbmc-datasets-by-seurat-and-sccatch"><i class="fa fa-check"></i><b>14.9</b> Infer cell types for PBMC datasets by Seurat and scCATCH</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><i class="fa fa-check"></i><b>15</b> Appendix C: analyzing scRNA-seq data by existing method III</a>
<ul>
<li class="chapter" data-level="15.1" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#preprocessing-2"><i class="fa fa-check"></i><b>15.1</b> Preprocessing</a></li>
<li class="chapter" data-level="15.2" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#cluster-cells-2"><i class="fa fa-check"></i><b>15.2</b> Cluster cells</a></li>
<li class="chapter" data-level="15.3" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#visualize-data-in-low-dimensional-space-2"><i class="fa fa-check"></i><b>15.3</b> Visualize data in low dimensional space</a></li>
<li class="chapter" data-level="15.4" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#find-differentially-expressed-genes-2"><i class="fa fa-check"></i><b>15.4</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="15.5" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#save-results-3"><i class="fa fa-check"></i><b>15.5</b> Save results</a></li>
<li class="chapter" data-level="15.6" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#infer-cell-types-for-pbmc-datasets-2"><i class="fa fa-check"></i><b>15.6</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><i class="fa fa-check"></i><b>16</b> Appendix D: analyzing scRNA-seq data by existing method IV</a>
<ul>
<li class="chapter" data-level="16.1" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#preprocessing-3"><i class="fa fa-check"></i><b>16.1</b> Preprocessing</a></li>
<li class="chapter" data-level="16.2" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#cluster-cells-3"><i class="fa fa-check"></i><b>16.2</b> Cluster cells</a></li>
<li class="chapter" data-level="16.3" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#find-differentially-expressed-genes-3"><i class="fa fa-check"></i><b>16.3</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="16.4" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#save-results-4"><i class="fa fa-check"></i><b>16.4</b> Save results</a></li>
<li class="chapter" data-level="16.5" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#infer-cell-types-for-pbmc-datasets-3"><i class="fa fa-check"></i><b>16.5</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="appendix-e-analyzing-scrna-seq-data-using-asurat.html"><a href="appendix-e-analyzing-scrna-seq-data-using-asurat.html"><i class="fa fa-check"></i><b>17</b> Appendix E: analyzing scRNA-seq data using ASURAT</a>
<ul>
<li class="chapter" data-level="17.1" data-path="appendix-e-analyzing-scrna-seq-data-using-asurat.html"><a href="appendix-e-analyzing-scrna-seq-data-using-asurat.html#infer-cell-types-for-pbmc-datasets-4"><i class="fa fa-check"></i><b>17.1</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html"><i class="fa fa-check"></i><b>18</b> Appendix F: creating figures</a>
<ul>
<li class="chapter" data-level="18.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#pbmcs-datasets"><i class="fa fa-check"></i><b>18.1</b> PBMCs datasets</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#multi-layered-analysis"><i class="fa fa-check"></i><b>18.1.1</b> Multi-layered analysis</a></li>
<li class="chapter" data-level="18.1.2" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#infer-population-ratios"><i class="fa fa-check"></i><b>18.1.2</b> Infer population ratios</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#lung-cancer-dataset"><i class="fa fa-check"></i><b>18.2</b> Lung cancer dataset</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#heatmaps-of-marker-gene-expressions"><i class="fa fa-check"></i><b>18.2.1</b> Heatmaps of marker gene expressions</a></li>
<li class="chapter" data-level="18.2.2" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#multi-layered-analysis-1"><i class="fa fa-check"></i><b>18.2.2</b> Multi-layered analysis</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#pancreatic-ductal-adenocarcinoma-pdac-datasets"><i class="fa fa-check"></i><b>18.3</b> Pancreatic ductal adenocarcinoma (PDAC) datasets</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#clustering-pdac-spatial-transcriptomes"><i class="fa fa-check"></i><b>18.3.1</b> Clustering PDAC spatial transcriptomes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to ASURAT (version 0.0.0.9001)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii" class="section level1" number="14">
<h1><span class="header-section-number">Chapter 14</span> Appendix B: analyzing scRNA-seq data by existing method II</h1>
<p>In this section, we analyze scRNA-seq datasets using Seurat.</p>
<p>The following libraries are used.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)       <span class="co"># For efficient handling of data.frame</span></span>
<span id="cb1-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)          <span class="co"># For using Seurat</span></span>
<span id="cb1-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(destiny)         <span class="co"># For using DiffusionMap()</span></span>
<span id="cb1-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)              <span class="co"># For using datatable</span></span>
<span id="cb1-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)    <span class="co"># For using human genome annotation package</span></span>
<span id="cb1-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)            <span class="co"># For using DB for Disease Ontology</span></span>
<span id="cb1-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler) <span class="co"># For using DB for Gene Ontology</span></span>
<span id="cb1-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GOSemSim)        <span class="co"># For using godata in GOSemSim package</span></span>
<span id="cb1-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scCATCH)         <span class="co"># For using scCATCH</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">packageVersion</span>(<span class="st">&quot;Seurat&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &#39;4.0.3&#39;</code></pre>
<pre><code>[1] &#39;4.0.3&#39;</code></pre>
<div id="preprocessing-1" class="section level2" number="14.1">
<h2><span class="header-section-number">14.1</span> Preprocessing</h2>
<p>Load the data.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span>
<span id="cb5-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_seurat.R&quot;</span>)</span>
<span id="cb5-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb5-3" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/01_005_sc68_vehi_normalized.rds&quot;</span>)</span>
<span id="cb5-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb5-4" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/02_005_sc68_cisp_normalized.rds&quot;</span>)</span>
<span id="cb5-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb5-5" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/10_005_pbmc_4000_normalized.rds&quot;</span>)</span>
<span id="cb5-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb5-6" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/11_005_pbmc_6000_normalized.rds&quot;</span>)</span></code></pre></div>
<p>Check the size of the gene-by-sample matrices.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(sc68_vehi[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]])</span>
<span id="cb6-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(sc68_cisp[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]])</span>
<span id="cb6-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(pbmc_4000[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]])</span>
<span id="cb6-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(pbmc_6000[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]])</span></code></pre></div>
<pre><code>[1] 6581 3923
[1] 6347 2285
[1] 6658 3815
[1] 5169 4878</code></pre>
<p>Create Seruat objects.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb8-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> sc68_vehi[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;bayNorm&quot;</span>]][[<span class="st">&quot;Bay_out&quot;</span>]], <span class="at">project =</span> <span class="st">&quot;sc68_vehi&quot;</span>)</span>
<span id="cb8-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-4" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb8-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> sc68_cisp[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;bayNorm&quot;</span>]][[<span class="st">&quot;Bay_out&quot;</span>]], <span class="at">project =</span> <span class="st">&quot;sc68_cisp&quot;</span>)</span>
<span id="cb8-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-7" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb8-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> pbmc_4000[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;bayNorm&quot;</span>]][[<span class="st">&quot;Bay_out&quot;</span>]], <span class="at">project =</span> <span class="st">&quot;pbmc_4000&quot;</span>)</span>
<span id="cb8-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-10" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb8-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> pbmc_6000[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;bayNorm&quot;</span>]][[<span class="st">&quot;Bay_out&quot;</span>]], <span class="at">project =</span> <span class="st">&quot;pbmc_6000&quot;</span>)</span></code></pre></div>
<p>According to <code>Seurat</code>â€™s protocol, we perform the following procedures:</p>
<ol style="list-style-type: decimal">
<li>normalize the data by using <code>NormalizeData()</code> in Seurat package
with the argument <code>normalization.method = LogNormalize</code>,</li>
<li>perform variance stabilizing transform (VST) by using
<code>FindVariableFeatures()</code> setting the number of variable feature <code>nfeatures</code>,</li>
<li>scale the data, and</li>
<li>reduce the dimension by principal component analysis (PCA),
using <code>RunPCA()</code> with the argument <code>features = VariableFeatures(obj)</code>.</li>
</ol>
<p><strong>Tips:</strong>
As mentioned in Cruz and Wishart, Cancer Inform. 2, 59-77 (2006),
we select highly variable genes as the cell-per-variable gene ratio being 5:1.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb9-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-4" aria-hidden="true" tabindex="-1"></a>nfeatures <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.2</span> <span class="sc">*</span> sc68_vehi<span class="sc">@</span>assays[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">@</span>counts<span class="sc">@</span>Dim[<span class="dv">2</span>])</span>
<span id="cb9-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-5" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">process_001_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">nfeatures =</span> nfeatures)</span>
<span id="cb9-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_cisp</span></span>
<span id="cb9-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-9" aria-hidden="true" tabindex="-1"></a>nfeatures <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.2</span> <span class="sc">*</span> sc68_cisp<span class="sc">@</span>assays[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">@</span>counts<span class="sc">@</span>Dim[<span class="dv">2</span>])</span>
<span id="cb9-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-10" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">process_001_seurat</span>(<span class="at">obj =</span> sc68_cisp, <span class="at">nfeatures =</span> nfeatures)</span>
<span id="cb9-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_4000</span></span>
<span id="cb9-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-14" aria-hidden="true" tabindex="-1"></a>nfeatures <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.2</span> <span class="sc">*</span> pbmc_4000<span class="sc">@</span>assays[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">@</span>counts<span class="sc">@</span>Dim[<span class="dv">2</span>])</span>
<span id="cb9-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-15" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">process_001_seurat</span>(<span class="at">obj =</span> pbmc_4000, <span class="at">nfeatures =</span> nfeatures)</span>
<span id="cb9-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_6000</span></span>
<span id="cb9-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb9-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-19" aria-hidden="true" tabindex="-1"></a>nfeatures <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.2</span> <span class="sc">*</span> pbmc_6000<span class="sc">@</span>assays[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">@</span>counts<span class="sc">@</span>Dim[<span class="dv">2</span>])</span>
<span id="cb9-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb9-20" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">process_001_seurat</span>(<span class="at">obj =</span> pbmc_6000, <span class="at">nfeatures =</span> nfeatures)</span></code></pre></div>
<p>Check the eigen value (variance) of each principal component by elbow plot.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb10-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb10-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb10-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot_pcaEigen_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">title =</span> <span class="st">&quot;sc68_vehi (Seurat)&quot;</span>,</span>
<span id="cb10-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">title_size =</span> <span class="dv">18</span>, <span class="at">xlabel =</span> <span class="st">&quot;PC&quot;</span>,</span>
<span id="cb10-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ylabel =</span> <span class="st">&quot;Eigen value&quot;</span>)</span>
<span id="cb10-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-7" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0600.png&quot;</span></span>
<span id="cb10-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="fl">3.5</span>, <span class="at">height =</span> <span class="fl">3.5</span>)</span></code></pre></div>
<p><img src="figures/figure_01_0600.png" width="250px">
<img src="figures/figure_02_0600.png" width="250px"></p>
<p><img src="figures/figure_10_0600.png" width="250px">
<img src="figures/figure_11_0600.png" width="250px"></p>
</div>
<div id="cluster-cells-1" class="section level2" number="14.2">
<h2><span class="header-section-number">14.2</span> Cluster cells</h2>
<p>Calculate the cumulative sum of variances.
These results are used to determine the dimensions of the principal components.
In what follows, we use the PCs whose contribution ratio is greater than
or equal to 0.9 for the first time.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi_pc <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">cumsum</span>(sc68_vehi<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">/</span></span>
<span id="cb11-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-2" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">sum</span>(sc68_vehi<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">&gt;</span> <span class="fl">0.9</span>)[<span class="dv">1</span>]</span>
<span id="cb11-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-3" aria-hidden="true" tabindex="-1"></a>sc68_cisp_pc <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">cumsum</span>(sc68_cisp<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">/</span></span>
<span id="cb11-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-4" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">sum</span>(sc68_cisp<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">&gt;</span> <span class="fl">0.9</span>)[<span class="dv">1</span>]</span>
<span id="cb11-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-5" aria-hidden="true" tabindex="-1"></a>pbmc_4000_pc <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">cumsum</span>(pbmc_4000<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">/</span></span>
<span id="cb11-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">sum</span>(pbmc_4000<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">&gt;</span> <span class="fl">0.9</span>)[<span class="dv">1</span>]</span>
<span id="cb11-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-7" aria-hidden="true" tabindex="-1"></a>pbmc_6000_pc <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">cumsum</span>(pbmc_6000<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">/</span></span>
<span id="cb11-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb11-8" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">sum</span>(pbmc_6000<span class="sc">@</span>reductions[[<span class="st">&quot;pca&quot;</span>]]<span class="sc">@</span>stdev) <span class="sc">&gt;</span> <span class="fl">0.9</span>)[<span class="dv">1</span>]</span></code></pre></div>
<p>According to <code>Seurat</code>â€™s protocol, we perform the following procedures:</p>
<ol style="list-style-type: decimal">
<li>run <code>FindNeighbors()</code> for creating k-nearest neighbor graph,</li>
<li>run <code>FindClusters()</code> for clustering cells, and</li>
<li>run <code>RunTSNE()</code> and <code>RunUMAP()</code></li>
</ol>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">process_002_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">pc =</span> sc68_vehi_pc,</span>
<span id="cb12-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.1</span>)</span>
<span id="cb12-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-3" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">process_002_seurat</span>(<span class="at">obj =</span> sc68_cisp, <span class="at">pc =</span> sc68_cisp_pc,</span>
<span id="cb12-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.1</span>)</span>
<span id="cb12-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-5" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">process_002_seurat</span>(<span class="at">obj =</span> pbmc_4000, <span class="at">pc =</span> pbmc_4000_pc,</span>
<span id="cb12-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.08</span>)</span>
<span id="cb12-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-7" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">process_002_seurat</span>(<span class="at">obj =</span> pbmc_6000, <span class="at">pc =</span> pbmc_6000_pc,</span>
<span id="cb12-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb12-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.15</span>)</span></code></pre></div>
<p>Change the label names by adding <code>1</code> (e.g., change from <code>0</code> to <code>1</code>).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb13-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-4" aria-hidden="true" tabindex="-1"></a>sc68_vehi<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">as.integer</span>(</span>
<span id="cb13-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>(sc68_vehi<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]])) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb13-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-6" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(sc68_vehi, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span>
<span id="cb13-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_cisp</span></span>
<span id="cb13-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-10" aria-hidden="true" tabindex="-1"></a>sc68_cisp<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">as.integer</span>(</span>
<span id="cb13-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>(sc68_cisp<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]])) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb13-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-12" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(sc68_cisp, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span>
<span id="cb13-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_4000</span></span>
<span id="cb13-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-16" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">as.integer</span>(</span>
<span id="cb13-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>(pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]])) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb13-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-18" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(pbmc_4000, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span>
<span id="cb13-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_6000</span></span>
<span id="cb13-21"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb13-22"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-22" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">as.integer</span>(</span>
<span id="cb13-23"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>(pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]])) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb13-24"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb13-24" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(pbmc_6000, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span></code></pre></div>
</div>
<div id="visualize-data-in-low-dimensional-space-1" class="section level2" number="14.3">
<h2><span class="header-section-number">14.3</span> Visualize data in low dimensional space</h2>
<p>The following function <code>plot_tsne_seurat()</code> shows clustering results in a
two-dimensional t-SNE space.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb14-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb14-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb14-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot_tsne_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">title =</span> <span class="st">&quot;sc68_vehi (Seurat)&quot;</span>,</span>
<span id="cb14-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">title_size =</span> <span class="dv">18</span>, <span class="at">xlabel =</span> <span class="st">&quot;tSNE_1&quot;</span>, <span class="at">ylabel =</span> <span class="st">&quot;tSNE_2&quot;</span>,</span>
<span id="cb14-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">default_color =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-7" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0605.png&quot;</span></span>
<span id="cb14-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="fl">4.8</span>, <span class="at">height =</span> <span class="fl">4.0</span>)</span></code></pre></div>
<p><img src="figures/figure_01_0605.png" width="250px">
<img src="figures/figure_02_0605.png" width="250px"></p>
<p><img src="figures/figure_10_0605.png" width="250px">
<img src="figures/figure_11_0605.png" width="250px"></p>
<p>The following function <code>plot_umap_seurat()</code> shows clustering results in a
two-dimensional UMAP space.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb15-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb15-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb15-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot_umap_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">title =</span> <span class="st">&quot;sc68_vehi (Seurat)&quot;</span>,</span>
<span id="cb15-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">title_size =</span> <span class="dv">18</span>, <span class="at">xlabel =</span> <span class="st">&quot;UMAP_1&quot;</span>, <span class="at">ylabel =</span> <span class="st">&quot;UMAP_2&quot;</span>,</span>
<span id="cb15-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">default_color =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-7" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0606.png&quot;</span></span>
<span id="cb15-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="fl">4.8</span>, <span class="at">height =</span> <span class="fl">4.0</span>)</span></code></pre></div>
<p><img src="figures/figure_01_0606.png" width="250px">
<img src="figures/figure_02_0606.png" width="250px"></p>
<p><img src="figures/figure_10_0606.png" width="250px">
<img src="figures/figure_11_0606.png" width="250px"></p>
<p>The following function <code>do_dmap_seurat()</code> performs <code>DiffusionMap()</code> using
<code>destiny</code> package, where the arguments are preset as <code>sigma = "local"</code> and
<code>distance = "euclidean"</code>.
The arguments are <code>obj</code>, <code>sigma</code> (argument of <code>DiffusionMap()</code>), <code>distance</code>
(argument of <code>DiffusionMap()</code>), and <code>pca_dim</code> (an integer: dimension of
principal component, in which <code>NULL</code> is accepted if users would like to compute
diffusion map from the original gene expression matrices).</p>
<p><strong>Tips:</strong>
If the resulting diffusion map is distorted by outliers, tuning of <code>pca_dim</code>
will be required.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb16-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">do_dmap_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">pca_dim =</span> sc68_vehi_pc)</span>
<span id="cb16-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb16-2" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">do_dmap_seurat</span>(<span class="at">obj =</span> sc68_cisp, <span class="at">pca_dim =</span> sc68_cisp_pc)</span>
<span id="cb16-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb16-3" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">do_dmap_seurat</span>(<span class="at">obj =</span> pbmc_4000, <span class="at">pca_dim =</span> pbmc_4000_pc)</span>
<span id="cb16-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb16-4" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">do_dmap_seurat</span>(<span class="at">obj =</span> pbmc_6000, <span class="at">pca_dim =</span> pbmc_6000_pc)</span></code></pre></div>
<p>The results are stored in <code>obj@reductions[["dmap"]]</code>.</p>
<p>The following function <code>plot_dmap_seurat()</code> shows the data in a two-dimensional
diffusion map.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb17-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb17-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb17-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot_dmap_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">title =</span> <span class="st">&quot;sc68_vehi (Seurat)&quot;</span>,</span>
<span id="cb17-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">title_size =</span> <span class="dv">18</span>, <span class="at">xlabel =</span> <span class="st">&quot;DC_1&quot;</span>, <span class="at">ylabel =</span> <span class="st">&quot;DC_2&quot;</span>,</span>
<span id="cb17-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">default_color =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-7" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0607.png&quot;</span></span>
<span id="cb17-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="fl">5.0</span>, <span class="at">height =</span> <span class="fl">4.0</span>)</span></code></pre></div>
<p><img src="figures/figure_01_0607.png" width="250px">
<img src="figures/figure_02_0607.png" width="250px"></p>
<p><img src="figures/figure_10_0607.png" width="250px">
<img src="figures/figure_11_0607.png" width="250px"></p>
</div>
<div id="cell-cycle-scoring" class="section level2" number="14.4">
<h2><span class="header-section-number">14.4</span> Cell cycle scoring</h2>
<p>Assign each cell a cell cycle score using <code>CellCycleScoring()</code>.
Note that the entries of <code>obj@active.ident</code> will be replaced with
<code>obj@meta.data[["Phase"]]</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-1" aria-hidden="true" tabindex="-1"></a>s.genes <span class="ot">&lt;-</span> cc.genes<span class="sc">$</span>s.genes</span>
<span id="cb18-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-2" aria-hidden="true" tabindex="-1"></a>g2m.genes <span class="ot">&lt;-</span> cc.genes<span class="sc">$</span>g2m.genes</span>
<span id="cb18-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-4" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(sc68_vehi, <span class="at">s.features =</span> s.genes,</span>
<span id="cb18-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">g2m.features =</span> g2m.genes, <span class="at">set.ident =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-6" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(sc68_cisp, <span class="at">s.features =</span> s.genes,</span>
<span id="cb18-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">g2m.features =</span> g2m.genes, <span class="at">set.ident =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-8" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(pbmc_4000, <span class="at">s.features =</span> s.genes,</span>
<span id="cb18-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-9" aria-hidden="true" tabindex="-1"></a>                              <span class="at">g2m.features =</span> g2m.genes, <span class="at">set.ident =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-10" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(pbmc_6000, <span class="at">s.features =</span> s.genes,</span>
<span id="cb18-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">g2m.features =</span> g2m.genes, <span class="at">set.ident =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-13" aria-hidden="true" tabindex="-1"></a>sc68_vehi<span class="sc">@</span>meta.data[[<span class="st">&quot;Phase&quot;</span>]] <span class="ot">&lt;-</span> sc68_vehi<span class="sc">@</span>active.ident</span>
<span id="cb18-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-14" aria-hidden="true" tabindex="-1"></a>sc68_cisp<span class="sc">@</span>meta.data[[<span class="st">&quot;Phase&quot;</span>]] <span class="ot">&lt;-</span> sc68_cisp<span class="sc">@</span>active.ident</span>
<span id="cb18-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-15" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;Phase&quot;</span>]] <span class="ot">&lt;-</span> pbmc_4000<span class="sc">@</span>active.ident</span>
<span id="cb18-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb18-16" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;Phase&quot;</span>]] <span class="ot">&lt;-</span> pbmc_6000<span class="sc">@</span>active.ident</span></code></pre></div>
<p>The following function <code>plot_cc_umap_seurat()</code> shows cell cycle phases in
a two-dimensional UMAP space.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb19-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb19-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb19-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot_cc_umap_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">title =</span> <span class="st">&quot;sc68_vehi (Seurat)&quot;</span>,</span>
<span id="cb19-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title_size =</span> <span class="dv">18</span>, <span class="at">xlabel =</span> <span class="st">&quot;UMAP_1&quot;</span>, <span class="at">ylabel =</span> <span class="st">&quot;UMAP_2&quot;</span>,</span>
<span id="cb19-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">default_color =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-7" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0608.png&quot;</span></span>
<span id="cb19-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="fl">5.2</span>, <span class="at">height =</span> <span class="fl">4.0</span>)</span></code></pre></div>
<p><img src="figures/figure_01_0608.png" width="250px">
<img src="figures/figure_02_0608.png" width="250px"></p>
<p><img src="figures/figure_10_0608.png" width="250px">
<img src="figures/figure_11_0608.png" width="250px"></p>
<p>The following function <code>plot_ccscore_umap_seurat()</code> shows S to G2M scores,
i.e., standardized scores of mean expression levels of phase marker genes,
in a two-dimensional UMAP space.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb20-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb20-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb20-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot_ccscore_umap_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">title =</span> <span class="st">&quot;sc68_vehi (Seurat)&quot;</span>,</span>
<span id="cb20-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">title_size =</span> <span class="dv">18</span>, <span class="at">xlabel =</span> <span class="st">&quot;UMAP_1&quot;</span>,</span>
<span id="cb20-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">ylabel =</span> <span class="st">&quot;UMAP_2&quot;</span>)</span>
<span id="cb20-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-7" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0609.png&quot;</span></span>
<span id="cb20-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="fl">5.1</span>, <span class="at">height =</span> <span class="fl">4.0</span>)</span></code></pre></div>
<p><img src="figures/figure_01_0609.png" width="250px">
<img src="figures/figure_02_0609.png" width="250px"></p>
<p>Reset <code>obj@active.ident</code> into <code>obj@meta.data[["seurat_clusters"]]</code>.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb21-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(sc68_vehi, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span>
<span id="cb21-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb21-2" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(sc68_cisp, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span>
<span id="cb21-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb21-3" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(pbmc_4000, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span>
<span id="cb21-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb21-4" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(pbmc_6000, <span class="at">value =</span> <span class="st">&quot;seurat_clusters&quot;</span>)</span></code></pre></div>
</div>
<div id="find-differentially-expressed-genes-1" class="section level2" number="14.5">
<h2><span class="header-section-number">14.5</span> Find differentially expressed genes</h2>
<p>Find differentially expressed genes using Seuratâ€™s function and stores the
results into <code>obj@misc[["markers"]]</code>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb22-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  sc68_vehi, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span>
<span id="cb22-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-4" aria-hidden="true" tabindex="-1"></a>sc68_cisp<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb22-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-5" aria-hidden="true" tabindex="-1"></a>  sc68_cisp, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span>
<span id="cb22-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-7" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb22-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-8" aria-hidden="true" tabindex="-1"></a>  pbmc_4000, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span>
<span id="cb22-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-10" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb22-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb22-11" aria-hidden="true" tabindex="-1"></a>  pbmc_6000, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span></code></pre></div>
<p>Users can check the marker genes of each cluster by the following command:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(sc68_vehi<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]], <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb23-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(sc68_cisp<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]], <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb23-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]], <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb23-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]], <span class="at">rownames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ul>
<li><code>sc68_vehi</code></li>
</ul>
<iframe src="figures/sc68_vehi_marker_seurat.html" style="width:100%; height:400px;">
</iframe>
<p><br><br><br><br><br><br><br><br><br><br></p>
<ul>
<li><code>sc68_cisp</code></li>
</ul>
<iframe src="figures/sc68_cisp_marker_seurat.html" style="width:100%; height:400px;">
</iframe>
<p><br><br><br><br><br><br><br><br><br><br></p>
<ul>
<li><code>pbmc_4000</code></li>
</ul>
<iframe src="figures/pbmc_4000_marker_seurat.html" style="width:100%; height:400px;">
</iframe>
<p><br><br><br><br><br><br><br><br><br><br></p>
<ul>
<li><code>pbmc_6000</code></li>
</ul>
<iframe src="figures/pbmc_6000_marker_seurat.html" style="width:100%; height:400px;">
</iframe>
<p><br><br><br><br><br><br><br><br><br><br></p>
</div>
<div id="enrichment-analysis" class="section level2" number="14.6">
<h2><span class="header-section-number">14.6</span> Enrichment analysis</h2>
<p>The following function <code>do_enrichGO_seurat()</code> performs <code>clusterProfiler</code>
functions as follows:</p>
<ol style="list-style-type: decimal">
<li><code>enrichGO()</code> for all Gene Ontology categories (MF, BP, CC, and ALL),</li>
<li><code>simplify()</code> to remove redundant GO terms (but only for MF, BP, and CC), and</li>
</ol>
<p>The arguments are <code>obj</code> (Seurat object), <code>padj_cutoff</code> (cutoff value of the
input data <code>obj@misc[["markers"]][["p_val_adj"]]</code>), <code>qval_cutoff_enrichGO</code>
(i.e., <code>qvalueCutoff</code>, an argument of <code>enrichGO()</code>), <code>orgdb</code>
(annotation data package such as <code>org.Hs.eg.db</code> and <code>org.Mm.eg.db</code>), and
<code>cutoff</code> (an argument of <code>simplify()</code>, ranging from 0 to 1).
The other arguments are set as <code>pAdjustMethod = "BH"</code>, <code>by = "p.adjust"</code>,
<code>select_fun = min</code>, <code>measure = "Wang"</code>, and <code>semData = NULL</code>.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">do_enrichGO_seurat</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb24-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">qval_cutoff_enrichGO =</span> <span class="fl">0.05</span>,</span>
<span id="cb24-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">orgdb =</span> org.Hs.eg.db, <span class="at">cutoff =</span> <span class="fl">0.7</span>,</span>
<span id="cb24-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>)</span>
<span id="cb24-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-5" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">do_enrichGO_seurat</span>(<span class="at">obj =</span> sc68_cisp, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb24-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">qval_cutoff_enrichGO =</span> <span class="fl">0.05</span>,</span>
<span id="cb24-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-7" aria-hidden="true" tabindex="-1"></a>                                <span class="at">orgdb =</span> org.Hs.eg.db, <span class="at">cutoff =</span> <span class="fl">0.7</span>,</span>
<span id="cb24-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>)</span>
<span id="cb24-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-9" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">do_enrichGO_seurat</span>(<span class="at">obj =</span> pbmc_4000, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb24-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-10" aria-hidden="true" tabindex="-1"></a>                                <span class="at">qval_cutoff_enrichGO =</span> <span class="fl">0.05</span>,</span>
<span id="cb24-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-11" aria-hidden="true" tabindex="-1"></a>                                <span class="at">orgdb =</span> org.Hs.eg.db, <span class="at">cutoff =</span> <span class="fl">0.7</span>,</span>
<span id="cb24-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-12" aria-hidden="true" tabindex="-1"></a>                                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>)</span>
<span id="cb24-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-13" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">do_enrichGO_seurat</span>(<span class="at">obj =</span> pbmc_6000, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb24-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">qval_cutoff_enrichGO =</span> <span class="fl">0.05</span>,</span>
<span id="cb24-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-15" aria-hidden="true" tabindex="-1"></a>                                <span class="at">orgdb =</span> org.Hs.eg.db, <span class="at">cutoff =</span> <span class="fl">0.7</span>,</span>
<span id="cb24-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb24-16" aria-hidden="true" tabindex="-1"></a>                                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>)</span></code></pre></div>
<p>The results are stored in <code>obj@misc[["enrichGO"]]</code> and
<code>obj@misc[["enrichGO_simplified"]]</code>.</p>
<p>The following function <code>prepare_glist()</code> prepares a list of genes, which is
used for an input of <code>compareCluster()</code>.
Note that we use <code>org.Hs.eg.db</code> for using human genome annotation package.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb25-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing</span></span>
<span id="cb25-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb25-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-4" aria-hidden="true" tabindex="-1"></a>prepare_glist <span class="ot">&lt;-</span> <span class="cf">function</span>(obj, padj_cutoff){</span>
<span id="cb25-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-5" aria-hidden="true" tabindex="-1"></a>  n_groups <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">sort</span>(obj<span class="sc">@</span>active.ident)))</span>
<span id="cb25-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-6" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> obj<span class="sc">@</span>misc[[<span class="st">&quot;markers&quot;</span>]]</span>
<span id="cb25-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-7" aria-hidden="true" tabindex="-1"></a>  cluster_names <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(tmp<span class="sc">$</span>cluster))</span>
<span id="cb25-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-8" aria-hidden="true" tabindex="-1"></a>  g <span class="ot">&lt;-</span> <span class="fu">list</span>() ; glist <span class="ot">&lt;-</span> <span class="fu">list</span>() ; label_names <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb25-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_groups){</span>
<span id="cb25-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-10" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> tmp[<span class="fu">which</span>(tmp<span class="sc">$</span>cluster <span class="sc">==</span> i),]</span>
<span id="cb25-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-11" aria-hidden="true" tabindex="-1"></a>    g[[i]] <span class="ot">&lt;-</span> df[<span class="fu">which</span>(df<span class="sc">$</span>p_val_adj <span class="sc">&lt;=</span> padj_cutoff),]<span class="sc">$</span>gene</span>
<span id="cb25-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-12" aria-hidden="true" tabindex="-1"></a>    geneID <span class="ot">&lt;-</span> <span class="fu">bitr</span>(g[[i]], <span class="at">fromType =</span> <span class="st">&quot;SYMBOL&quot;</span>, <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb25-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">OrgDb =</span> org.Hs.eg.db)<span class="sc">$</span>ENTREZID</span>
<span id="cb25-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-14" aria-hidden="true" tabindex="-1"></a>    glist[[i]] <span class="ot">&lt;-</span> geneID</span>
<span id="cb25-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-15" aria-hidden="true" tabindex="-1"></a>    label_names <span class="ot">&lt;-</span> <span class="fu">c</span>(label_names, <span class="fu">paste</span>(<span class="st">&quot;Group_&quot;</span>, i, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb25-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb25-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(glist) <span class="ot">&lt;-</span> label_names</span>
<span id="cb25-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(glist)</span>
<span id="cb25-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-21"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-22"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-22" aria-hidden="true" tabindex="-1"></a>glist_sc68_vehi <span class="ot">&lt;-</span> <span class="fu">prepare_glist</span>(<span class="at">obj =</span> sc68_vehi, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb25-23"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-23" aria-hidden="true" tabindex="-1"></a>glist_sc68_cisp <span class="ot">&lt;-</span> <span class="fu">prepare_glist</span>(<span class="at">obj =</span> sc68_cisp, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb25-24"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-24" aria-hidden="true" tabindex="-1"></a>glist_pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">prepare_glist</span>(<span class="at">obj =</span> pbmc_4000, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb25-25"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb25-25" aria-hidden="true" tabindex="-1"></a>glist_pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">prepare_glist</span>(<span class="at">obj =</span> pbmc_6000, <span class="at">padj_cutoff =</span> <span class="fl">0.01</span>)</span></code></pre></div>
<p>Performs <code>compareCluster()</code>, which easily compares enriched biological terms
across clusters (but without <code>simplify()</code>).</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb26-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-4" aria-hidden="true" tabindex="-1"></a>sc68_vehi<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_GO&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-5" aria-hidden="true" tabindex="-1"></a>  glist_sc68_vehi, <span class="at">fun =</span> <span class="st">&quot;enrichGO&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb26-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb26-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-8" aria-hidden="true" tabindex="-1"></a>sc68_vehi<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_KEGG&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-9" aria-hidden="true" tabindex="-1"></a>  glist_sc68_vehi, <span class="at">fun =</span> <span class="st">&quot;enrichKEGG&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>, <span class="at">keyType =</span> <span class="st">&quot;kegg&quot;</span>,</span>
<span id="cb26-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb26-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">500</span>) <span class="co"># min/max size of genes annotated for testing</span></span>
<span id="cb26-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_cisp</span></span>
<span id="cb26-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-15" aria-hidden="true" tabindex="-1"></a>sc68_cisp<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_GO&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-16" aria-hidden="true" tabindex="-1"></a>  glist_sc68_cisp, <span class="at">fun =</span> <span class="st">&quot;enrichGO&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb26-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb26-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-19" aria-hidden="true" tabindex="-1"></a>sc68_cisp<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_KEGG&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-20" aria-hidden="true" tabindex="-1"></a>  glist_sc68_cisp, <span class="at">fun =</span> <span class="st">&quot;enrichKEGG&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>, <span class="at">keyType =</span> <span class="st">&quot;kegg&quot;</span>,</span>
<span id="cb26-21"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb26-22"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">500</span>) <span class="co"># min/max size of genes annotated for testing</span></span>
<span id="cb26-23"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-24"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_4000</span></span>
<span id="cb26-25"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-26"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-26" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_GO&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-27"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-27" aria-hidden="true" tabindex="-1"></a>  glist_pbmc_4000, <span class="at">fun =</span> <span class="st">&quot;enrichGO&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb26-28"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb26-29"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-30"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-30" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_KEGG&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-31"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-31" aria-hidden="true" tabindex="-1"></a>  glist_pbmc_4000, <span class="at">fun =</span> <span class="st">&quot;enrichKEGG&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>, <span class="at">keyType =</span> <span class="st">&quot;kegg&quot;</span>,</span>
<span id="cb26-32"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb26-33"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">500</span>) <span class="co"># min/max size of genes annotated for testing</span></span>
<span id="cb26-34"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-34" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-35"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-35" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_6000</span></span>
<span id="cb26-36"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-36" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb26-37"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-37" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_GO&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-38"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-38" aria-hidden="true" tabindex="-1"></a>  glist_pbmc_6000, <span class="at">fun =</span> <span class="st">&quot;enrichGO&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb26-39"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb26-40"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-41"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-41" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_KEGG&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb26-42"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-42" aria-hidden="true" tabindex="-1"></a>  glist_pbmc_6000, <span class="at">fun =</span> <span class="st">&quot;enrichKEGG&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>, <span class="at">keyType =</span> <span class="st">&quot;kegg&quot;</span>,</span>
<span id="cb26-43"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.01</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb26-44"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb26-44" aria-hidden="true" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">500</span>) <span class="co"># min/max size of genes annotated for testing</span></span></code></pre></div>
<p>Plot the results of <code>compareCluster()</code>.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb27-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sc68_vehi</span></span>
<span id="cb27-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co">#--------------------------------------------------</span></span>
<span id="cb27-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(sc68_cisp<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_GO&quot;</span>]], <span class="at">showCategory =</span> <span class="dv">5</span>)</span>
<span id="cb27-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-5" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0610.png&quot;</span></span>
<span id="cb27-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb27-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-8" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(sc68_cisp<span class="sc">@</span>misc[[<span class="st">&quot;compareCluster_KEGG&quot;</span>]], <span class="at">showCategory =</span> <span class="dv">8</span>)</span>
<span id="cb27-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-9" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">&quot;figures/figure_01_0611.png&quot;</span></span>
<span id="cb27-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> filename, <span class="at">plot =</span> p, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<ul>
<li><code>sc68_vehi</code></li>
</ul>
<p><img src="figures/figure_01_0610.png" width="400px">
<img src="figures/figure_01_0611.png" width="400px"></p>
<ul>
<li><code>sc68_cisp</code></li>
</ul>
<p><img src="figures/figure_02_0610.png" width="400px">
<img src="figures/figure_02_0611.png" width="400px"></p>
<ul>
<li><code>pbmc_4000</code></li>
</ul>
<p><img src="figures/figure_10_0610.png" width="400px">
<img src="figures/figure_10_0611.png" width="400px"></p>
<ul>
<li><code>pbmc_6000</code></li>
</ul>
<p><img src="figures/figure_11_0610.png" width="400px">
<img src="figures/figure_11_0611.png" width="400px"></p>
</div>
<div id="save-results-2" class="section level2" number="14.7">
<h2><span class="header-section-number">14.7</span> Save results</h2>
<p>Save the objects.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sc68_vehi, <span class="at">file =</span> <span class="st">&quot;backup/01_600_sc68_vehi_seurat.rds&quot;</span>)</span>
<span id="cb28-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sc68_cisp, <span class="at">file =</span> <span class="st">&quot;backup/02_600_sc68_cisp_seurat.rds&quot;</span>)</span>
<span id="cb28-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_4000, <span class="at">file =</span> <span class="st">&quot;backup/10_600_pbmc_4000_seurat.rds&quot;</span>)</span>
<span id="cb28-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_6000, <span class="at">file =</span> <span class="st">&quot;backup/11_600_pbmc_6000_seurat.rds&quot;</span>)</span></code></pre></div>
<p>One can load the results.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb29-1" aria-hidden="true" tabindex="-1"></a>sc68_vehi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/01_600_sc68_vehi_seurat.rds&quot;</span>)</span>
<span id="cb29-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb29-2" aria-hidden="true" tabindex="-1"></a>sc68_cisp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/02_600_sc68_cisp_seurat.rds&quot;</span>)</span>
<span id="cb29-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb29-3" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/10_600_pbmc_4000_seurat.rds&quot;</span>)</span>
<span id="cb29-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb29-4" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/11_600_pbmc_6000_seurat.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="infer-cell-types-for-pbmc-datasets-1" class="section level2" number="14.8">
<h2><span class="header-section-number">14.8</span> Infer cell types for PBMC datasets</h2>
<p>Load the data.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span>
<span id="cb30-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_seurat.R&quot;</span>)</span>
<span id="cb30-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb30-3" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/10_600_pbmc_4000_seurat.rds&quot;</span>)</span>
<span id="cb30-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb30-4" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/11_600_pbmc_6000_seurat.rds&quot;</span>)</span></code></pre></div>
<p>Based on the result of <code>FindAllMarkers()</code>, we manually investigate marker genes
using GeneCards as follows:</p>
<ul>
<li><code>pbmc_4000</code></li>
</ul>
<pre><code>1: T cell         # TRAC (FDR ~ 0), CD3D (FDR ~ e-251), IL32 (FDR ~ e-131)
2: Monocyte       # S100A8 (FDR ~ 0), LYZ (FDR ~ 0), CD14 (FDR ~ e-119)
3: B cell         # CD79A (FDR ~ e-301), MS4A1 (FDR ~ e-276), IGHM (FDR ~ e-234)
4: NK/NKT cell    # CD3D (FDR ~ e-211), NKG7 (FDR ~ e-211   ), GZMA (FDR ~ e-198)
5: NK/NKT cell    # KLRF1 (FDR ~ 0), GZMB (FDR ~ e-264), CD160 (FDR ~ e-159)
6: Unspecified    # TCF4, MZB1, PTCRA, JCHAIN, MZB1, GZMB, ..</code></pre>
<ul>
<li><code>pbmc_6000</code></li>
</ul>
<pre><code>1: T cell         # CD3D (FDR ~ 0), TCF7 (FDR ~ 0), CD27 (FDR ~ e-225)
2: Monocyte       # S100A8 (FDR ~ 0), LYZ (FDR ~ 0), CD14 (FDR ~ 0)
3: NK/NKT cell    # NKG7 (FDR ~ 0), GZMA (FDR ~ 0), FGFBP2 (FDR ~ 0)
4: B cell         # CD79A (FDR ~ 0), VPREB3 (FDR ~ 0), BANK1 (FDR ~ 0)</code></pre>
<p>Since our manual annotations are more comparable with those of previous research
(Cao et al., Front Genet, 2020), we identify the cell types as follows:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb33-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_4000</span></span>
<span id="cb33-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb33-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-4" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.character</span>(pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]]))</span>
<span id="cb33-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-5" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;T cell&quot;</span></span>
<span id="cb33-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-6" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Monocyte&quot;</span></span>
<span id="cb33-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-7" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B cell&quot;</span></span>
<span id="cb33-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-8" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;NK or NKT cell&quot;</span></span>
<span id="cb33-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-9" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="st">&quot;NK or NKT cell&quot;</span></span>
<span id="cb33-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-10" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Unspecified&quot;</span></span>
<span id="cb33-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-11" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">factor</span>(tmp, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;T cell&quot;</span>, <span class="st">&quot;Monocyte&quot;</span>, <span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;NK or NKT cell&quot;</span>,</span>
<span id="cb33-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-12" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Unspecified&quot;</span>))</span>
<span id="cb33-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-13" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;mylabel&quot;</span>]] <span class="ot">&lt;-</span> tmp</span>
<span id="cb33-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb33-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co"># For `pbmc_6000`</span></span>
<span id="cb33-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb33-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-17" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.character</span>(pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]]))</span>
<span id="cb33-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-18" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;T cell&quot;</span></span>
<span id="cb33-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-19" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Monocyte&quot;</span></span>
<span id="cb33-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-20" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;NK or NKT cell&quot;</span></span>
<span id="cb33-21"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-21" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B cell&quot;</span></span>
<span id="cb33-22"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-22" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">factor</span>(tmp, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;T cell&quot;</span>, <span class="st">&quot;Monocyte&quot;</span>, <span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;NK or NKT cell&quot;</span>))</span>
<span id="cb33-23"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb33-23" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;mylabel&quot;</span>]] <span class="ot">&lt;-</span> tmp</span></code></pre></div>
<p>The numbers of cells are stored in a table.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-1" aria-hidden="true" tabindex="-1"></a>identify_cell <span class="ot">&lt;-</span> <span class="cf">function</span>(obj){</span>
<span id="cb34-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> obj<span class="sc">@</span>meta.data[[<span class="st">&quot;mylabel&quot;</span>]]</span>
<span id="cb34-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-3" aria-hidden="true" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">levels</span>(tmp)</span>
<span id="cb34-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-4" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb34-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(cell <span class="cf">in</span> cells){</span>
<span id="cb34-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-6" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df, <span class="fu">c</span>(cell, <span class="fu">length</span>(tmp[tmp<span class="sc">==</span>cell])))</span>
<span id="cb34-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-8" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb34-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;n&quot;</span>)</span>
<span id="cb34-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb34-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-13" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;population&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">identify_cell</span>(<span class="at">obj =</span> pbmc_4000)</span>
<span id="cb34-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb34-14" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;population&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">identify_cell</span>(<span class="at">obj =</span> pbmc_6000)</span></code></pre></div>
<p>Save the objects.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_4000, <span class="at">file =</span> <span class="st">&quot;backup/10_601_pbmc_4000_seurat.rds&quot;</span>)</span>
<span id="cb35-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_6000, <span class="at">file =</span> <span class="st">&quot;backup/11_601_pbmc_6000_seurat.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="infer-cell-types-for-pbmc-datasets-by-seurat-and-sccatch" class="section level2" number="14.9">
<h2><span class="header-section-number">14.9</span> Infer cell types for PBMC datasets by Seurat and scCATCH</h2>
<p>Load the data.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span>
<span id="cb36-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_seurat.R&quot;</span>)</span>
<span id="cb36-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb36-3" aria-hidden="true" tabindex="-1"></a>pbmc_4000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/10_600_pbmc_4000_seurat.rds&quot;</span>)</span>
<span id="cb36-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb36-4" aria-hidden="true" tabindex="-1"></a>pbmc_6000 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;backup/11_600_pbmc_6000_seurat.rds&quot;</span>)</span></code></pre></div>
<p>To make sure our manual annotations, we perform <code>findmarkergenes()</code> and
<code>scCATCH()</code> using <code>scCATCH</code> package.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb37-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_4000</span></span>
<span id="cb37-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb37-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-4" aria-hidden="true" tabindex="-1"></a>pbmc_4000_clu_markers <span class="ot">&lt;-</span> <span class="fu">findmarkergenes</span>(<span class="at">object =</span> pbmc_4000,</span>
<span id="cb37-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">species =</span> <span class="st">&quot;Human&quot;</span>,</span>
<span id="cb37-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cluster =</span> <span class="st">&#39;All&#39;</span>,</span>
<span id="cb37-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">match_CellMatch =</span> <span class="cn">FALSE</span>,</span>
<span id="cb37-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cancer =</span> <span class="cn">NULL</span>,</span>
<span id="cb37-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">tissue =</span> <span class="cn">NULL</span>,</span>
<span id="cb37-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cell_min_pct =</span> <span class="fl">0.25</span>,</span>
<span id="cb37-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">logfc =</span> <span class="fl">0.25</span>,</span>
<span id="cb37-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-12" aria-hidden="true" tabindex="-1"></a>                               <span class="at">pvalue =</span> <span class="fl">0.05</span>)</span>
<span id="cb37-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-13" aria-hidden="true" tabindex="-1"></a>pbmc_4000_clu_ann <span class="ot">&lt;-</span> <span class="fu">scCATCH</span>(<span class="at">object =</span> pbmc_4000_clu_markers<span class="sc">$</span>clu_markers,</span>
<span id="cb37-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-14" aria-hidden="true" tabindex="-1"></a>                             <span class="at">species =</span> <span class="st">&quot;Human&quot;</span>, <span class="at">cancer =</span> <span class="cn">NULL</span>,</span>
<span id="cb37-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-15" aria-hidden="true" tabindex="-1"></a>                             <span class="at">tissue =</span> <span class="st">&quot;Peripheral blood&quot;</span>)</span>
<span id="cb37-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-16" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;scCATCH&quot;</span>]][[<span class="st">&quot;clu_markers&quot;</span>]] <span class="ot">&lt;-</span> pbmc_4000_clu_markers</span>
<span id="cb37-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-17" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;scCATCH&quot;</span>]][[<span class="st">&quot;clu_ann&quot;</span>]] <span class="ot">&lt;-</span> pbmc_4000_clu_ann</span>
<span id="cb37-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-18" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb37-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-19" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_6000</span></span>
<span id="cb37-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-20" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb37-21"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-21" aria-hidden="true" tabindex="-1"></a>pbmc_6000_clu_markers <span class="ot">&lt;-</span> <span class="fu">findmarkergenes</span>(<span class="at">object =</span> pbmc_6000,</span>
<span id="cb37-22"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">species =</span> <span class="st">&quot;Human&quot;</span>,</span>
<span id="cb37-23"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-23" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cluster =</span> <span class="st">&#39;All&#39;</span>,</span>
<span id="cb37-24"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-24" aria-hidden="true" tabindex="-1"></a>                               <span class="at">match_CellMatch =</span> <span class="cn">FALSE</span>,</span>
<span id="cb37-25"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-25" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cancer =</span> <span class="cn">NULL</span>,</span>
<span id="cb37-26"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-26" aria-hidden="true" tabindex="-1"></a>                               <span class="at">tissue =</span> <span class="cn">NULL</span>,</span>
<span id="cb37-27"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-27" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cell_min_pct =</span> <span class="fl">0.25</span>,</span>
<span id="cb37-28"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-28" aria-hidden="true" tabindex="-1"></a>                               <span class="at">logfc =</span> <span class="fl">0.25</span>,</span>
<span id="cb37-29"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-29" aria-hidden="true" tabindex="-1"></a>                               <span class="at">pvalue =</span> <span class="fl">0.05</span>)</span>
<span id="cb37-30"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-30" aria-hidden="true" tabindex="-1"></a>pbmc_6000_clu_ann <span class="ot">&lt;-</span> <span class="fu">scCATCH</span>(<span class="at">object =</span> pbmc_6000_clu_markers<span class="sc">$</span>clu_markers,</span>
<span id="cb37-31"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-31" aria-hidden="true" tabindex="-1"></a>                             <span class="at">species =</span> <span class="st">&quot;Human&quot;</span>, <span class="at">cancer =</span> <span class="cn">NULL</span>,</span>
<span id="cb37-32"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-32" aria-hidden="true" tabindex="-1"></a>                             <span class="at">tissue =</span> <span class="st">&quot;Peripheral blood&quot;</span>)</span>
<span id="cb37-33"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-33" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;scCATCH&quot;</span>]][[<span class="st">&quot;clu_markers&quot;</span>]] <span class="ot">&lt;-</span> pbmc_6000_clu_markers</span>
<span id="cb37-34"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb37-34" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;scCATCH&quot;</span>]][[<span class="st">&quot;clu_ann&quot;</span>]] <span class="ot">&lt;-</span> pbmc_6000_clu_ann</span></code></pre></div>
<p>Note that the results of <code>scCATCH()</code> are different from
those of manual annotations.</p>
<p>Based on the result of <code>scCATCH()</code>, we annotated the cell type labels
as follows:</p>
<ul>
<li><code>pbmc_4000</code></li>
</ul>
<pre><code>1: Unspecified
2: Monocyte
3: B cell
4: T cell
5: Unspecified
6: T cell</code></pre>
<ul>
<li><code>pbmc_6000</code></li>
</ul>
<pre><code>1: T cell
2: Monocyte
3: T cell
4: B cell</code></pre>
<p>Since our manual annotations are more comparable with those of previous research
(Cao et al., Front Genet, 2020), we identify the cell types as follows:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb40-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_4000</span></span>
<span id="cb40-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb40-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-4" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.character</span>(pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]]))</span>
<span id="cb40-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-5" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Unspecified&quot;</span></span>
<span id="cb40-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-6" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Monocyte&quot;</span></span>
<span id="cb40-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-7" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B cell&quot;</span></span>
<span id="cb40-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-8" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;T cell&quot;</span></span>
<span id="cb40-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-9" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Unspecified&quot;</span></span>
<span id="cb40-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-10" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="st">&quot;T cell&quot;</span></span>
<span id="cb40-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-11" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">factor</span>(tmp, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;T cell&quot;</span>, <span class="st">&quot;Monocyte&quot;</span>, <span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;NK or NKT cell&quot;</span>,</span>
<span id="cb40-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-12" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;Unspecified&quot;</span>))</span>
<span id="cb40-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-13" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>meta.data[[<span class="st">&quot;mylabel&quot;</span>]] <span class="ot">&lt;-</span> tmp</span>
<span id="cb40-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb40-15"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co"># For `pbmc_6000`</span></span>
<span id="cb40-16"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ----------------------------------------</span></span>
<span id="cb40-17"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-17" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.character</span>(pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;seurat_clusters&quot;</span>]]))</span>
<span id="cb40-18"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-18" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;T cell&quot;</span></span>
<span id="cb40-19"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-19" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Monocyte&quot;</span></span>
<span id="cb40-20"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-20" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;NK or NKT cell&quot;</span></span>
<span id="cb40-21"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-21" aria-hidden="true" tabindex="-1"></a>tmp[tmp <span class="sc">==</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B cell&quot;</span></span>
<span id="cb40-22"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-22" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">factor</span>(tmp, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;T cell&quot;</span>, <span class="st">&quot;Monocyte&quot;</span>, <span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;NK or NKT cell&quot;</span>))</span>
<span id="cb40-23"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb40-23" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>meta.data[[<span class="st">&quot;mylabel&quot;</span>]] <span class="ot">&lt;-</span> tmp</span></code></pre></div>
<p>The numbers of cells are stored in a table.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-1" aria-hidden="true" tabindex="-1"></a>identify_cell <span class="ot">&lt;-</span> <span class="cf">function</span>(obj){</span>
<span id="cb41-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> obj<span class="sc">@</span>meta.data[[<span class="st">&quot;mylabel&quot;</span>]]</span>
<span id="cb41-3"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-3" aria-hidden="true" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">levels</span>(tmp)</span>
<span id="cb41-4"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-4" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb41-5"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(cell <span class="cf">in</span> cells){</span>
<span id="cb41-6"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-6" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df, <span class="fu">c</span>(cell, <span class="fu">length</span>(tmp[tmp<span class="sc">==</span>cell])))</span>
<span id="cb41-7"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-8"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-8" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb41-9"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;n&quot;</span>)</span>
<span id="cb41-10"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb41-11"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-12"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-13"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-13" aria-hidden="true" tabindex="-1"></a>pbmc_4000<span class="sc">@</span>misc[[<span class="st">&quot;population&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">identify_cell</span>(<span class="at">obj =</span> pbmc_4000)</span>
<span id="cb41-14"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb41-14" aria-hidden="true" tabindex="-1"></a>pbmc_6000<span class="sc">@</span>misc[[<span class="st">&quot;population&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">identify_cell</span>(<span class="at">obj =</span> pbmc_6000)</span></code></pre></div>
<p>Save the objects.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_4000, <span class="at">file =</span> <span class="st">&quot;backup/10_602_pbmc_4000_seurat_sccatch.rds&quot;</span>)</span>
<span id="cb42-2"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pbmc_6000, <span class="at">file =</span> <span class="st">&quot;backup/11_602_pbmc_6000_seurat_sccatch.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to ASURAT (ver. 0.0.0.9000).pdf", "Introduction to ASURAT (ver. 0.0.0.9000).epub"],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
