<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Collecting databases (optional) | Introduction to ASURAT (version 0.0.0.9001)</title>
  <meta name="description" content="This is a developmental version of ASURAT." />
  <meta name="generator" content="bookdown 0.23 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Collecting databases (optional) | Introduction to ASURAT (version 0.0.0.9001)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a developmental version of ASURAT." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Collecting databases (optional) | Introduction to ASURAT (version 0.0.0.9001)" />
  
  <meta name="twitter:description" content="This is a developmental version of ASURAT." />
  

<meta name="author" content="Keita Iida" />


<meta name="date" content="2021-09-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="markergenes.html"/>
<link rel="next" href="cell_type.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#license-and-description"><i class="fa fa-check"></i><b>1.1</b> License and description</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computational-environment"><i class="fa fa-check"></i><b>1.2</b> Computational environment</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dataset.html"><a href="dataset.html"><i class="fa fa-check"></i><b>2</b> Preparing data sets</a>
<ul>
<li class="chapter" data-level="2.1" data-path="dataset.html"><a href="dataset.html#case-i-10x-data-stewart-et-al.-nat-cancer-2020"><i class="fa fa-check"></i><b>2.1</b> Case I: 10x data (Stewart et al., Nat Cancer, 2020)</a></li>
<li class="chapter" data-level="2.2" data-path="dataset.html"><a href="dataset.html#case-ii-pdac-a-st-and-indrop-moncada-et-al.-nat-biotechnol-2020"><i class="fa fa-check"></i><b>2.2</b> Case II: PDAC-A ST and inDrop (Moncada et al., Nat Biotechnol, 2020)</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="dataset.html"><a href="dataset.html#pdac-a-st"><i class="fa fa-check"></i><b>2.2.1</b> PDAC-A ST</a></li>
<li class="chapter" data-level="2.2.2" data-path="dataset.html"><a href="dataset.html#pdac-a-indrop"><i class="fa fa-check"></i><b>2.2.2</b> PDAC-A inDrop</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="dataset.html"><a href="dataset.html#case-iii-input-10x-pbmc-data"><i class="fa fa-check"></i><b>2.3</b> Case III: input 10x PBMC data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataqc.html"><a href="dataqc.html"><i class="fa fa-check"></i><b>3</b> Data quality control (QC)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="dataqc.html"><a href="dataqc.html#quick-qc"><i class="fa fa-check"></i><b>3.1</b> Quick QC</a></li>
<li class="chapter" data-level="3.2" data-path="dataqc.html"><a href="dataqc.html#sample-qc"><i class="fa fa-check"></i><b>3.2</b> Sample QC</a></li>
<li class="chapter" data-level="3.3" data-path="dataqc.html"><a href="dataqc.html#variable-qc"><i class="fa fa-check"></i><b>3.3</b> Variable QC</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>4</b> Normalizing and centering data</a></li>
<li class="chapter" data-level="5" data-path="integration.html"><a href="integration.html"><i class="fa fa-check"></i><b>5</b> Data integration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="integration.html"><a href="integration.html#integrate-pdac-indrop-and-st-data-by-seurat"><i class="fa fa-check"></i><b>5.1</b> Integrate PDAC inDrop and ST data by Seurat</a></li>
<li class="chapter" data-level="5.2" data-path="integration.html"><a href="integration.html#identifying-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.2</b> Identifying differentially expressed genes</a></li>
<li class="chapter" data-level="5.3" data-path="integration.html"><a href="integration.html#inferring-cell-cycle-phases"><i class="fa fa-check"></i><b>5.3</b> Inferring cell cycle phases</a></li>
<li class="chapter" data-level="5.4" data-path="integration.html"><a href="integration.html#save-results"><i class="fa fa-check"></i><b>5.4</b> Save results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="computcorr.html"><a href="computcorr.html"><i class="fa fa-check"></i><b>6</b> Computing correlations among genes</a></li>
<li class="chapter" data-level="7" data-path="markergenes.html"><a href="markergenes.html"><i class="fa fa-check"></i><b>7</b> Checking expression profiles of marker genes</a></li>
<li class="chapter" data-level="8" data-path="collectdb.html"><a href="collectdb.html"><i class="fa fa-check"></i><b>8</b> Collecting databases (optional)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="collectdb.html"><a href="collectdb.html#disease-ontology-db"><i class="fa fa-check"></i><b>8.1</b> Disease Ontology DB</a></li>
<li class="chapter" data-level="8.2" data-path="collectdb.html"><a href="collectdb.html#cell-ontology-db"><i class="fa fa-check"></i><b>8.2</b> Cell Ontology DB</a></li>
<li class="chapter" data-level="8.3" data-path="collectdb.html"><a href="collectdb.html#gene-ontology-db"><i class="fa fa-check"></i><b>8.3</b> Gene Ontology DB</a></li>
<li class="chapter" data-level="8.4" data-path="collectdb.html"><a href="collectdb.html#kegg"><i class="fa fa-check"></i><b>8.4</b> KEGG</a></li>
<li class="chapter" data-level="8.5" data-path="collectdb.html"><a href="collectdb.html#custom-built-db-for-our-analysis-of-pdac-data"><i class="fa fa-check"></i><b>8.5</b> Custom-built DB for our analysis of PDAC data</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="cell_type.html"><a href="cell_type.html"><i class="fa fa-check"></i><b>9</b> Inferring cell types or diseases (optional)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="cell_type.html"><a href="cell_type.html#add-database-to-asurat-objects"><i class="fa fa-check"></i><b>9.1</b> Add database to ASURAT objects</a></li>
<li class="chapter" data-level="9.2" data-path="cell_type.html"><a href="cell_type.html#quick-qc-1"><i class="fa fa-check"></i><b>9.2</b> Quick QC</a></li>
<li class="chapter" data-level="9.3" data-path="cell_type.html"><a href="cell_type.html#correlation-graph-based-decomposition"><i class="fa fa-check"></i><b>9.3</b> Correlation graph-based decomposition</a></li>
<li class="chapter" data-level="9.4" data-path="cell_type.html"><a href="cell_type.html#creating-systems-of-signs"><i class="fa fa-check"></i><b>9.4</b> Creating systems of signs</a></li>
<li class="chapter" data-level="9.5" data-path="cell_type.html"><a href="cell_type.html#removing-redundant-signs"><i class="fa fa-check"></i><b>9.5</b> Removing redundant signs</a></li>
<li class="chapter" data-level="9.6" data-path="cell_type.html"><a href="cell_type.html#creating-sign-by-sample-matrices"><i class="fa fa-check"></i><b>9.6</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="9.7" data-path="cell_type.html"><a href="cell_type.html#dimensionality-reduction"><i class="fa fa-check"></i><b>9.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="cell_type.html"><a href="cell_type.html#principal-component-analysis-optional"><i class="fa fa-check"></i><b>9.7.1</b> Principal component analysis (optional)</a></li>
<li class="chapter" data-level="9.7.2" data-path="cell_type.html"><a href="cell_type.html#t-distributed-stochastic-neighbor-embedding-optional"><i class="fa fa-check"></i><b>9.7.2</b> t-distributed stochastic neighbor embedding (optional)</a></li>
<li class="chapter" data-level="9.7.3" data-path="cell_type.html"><a href="cell_type.html#uniform-manifold-approximation-and-projection-optional"><i class="fa fa-check"></i><b>9.7.3</b> Uniform manifold approximation and projection (optional)</a></li>
<li class="chapter" data-level="9.7.4" data-path="cell_type.html"><a href="cell_type.html#diffusion-map-optional"><i class="fa fa-check"></i><b>9.7.4</b> Diffusion map (optional)</a></li>
<li class="chapter" data-level="9.7.5" data-path="cell_type.html"><a href="cell_type.html#visualization"><i class="fa fa-check"></i><b>9.7.5</b> Visualization</a></li>
<li class="chapter" data-level="9.7.6" data-path="cell_type.html"><a href="cell_type.html#save-results."><i class="fa fa-check"></i><b>9.7.6</b> Save results.</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="cell_type.html"><a href="cell_type.html#cell-clustering"><i class="fa fa-check"></i><b>9.8</b> Cell clustering</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="cell_type.html"><a href="cell_type.html#partitioning-around-medoids-pam-optional"><i class="fa fa-check"></i><b>9.8.1</b> Partitioning around medoids (PAM) (optional)</a></li>
<li class="chapter" data-level="9.8.2" data-path="cell_type.html"><a href="cell_type.html#hierarchical-clustering-and-cutting-tree-optional"><i class="fa fa-check"></i><b>9.8.2</b> Hierarchical clustering and cutting tree (optional)</a></li>
<li class="chapter" data-level="9.8.3" data-path="cell_type.html"><a href="cell_type.html#community-detection-algorithm-with-k-nearest-neighbor-knn-graph-optional"><i class="fa fa-check"></i><b>9.8.3</b> Community detection algorithm with k-nearest neighbor (KNN) graph (optional)</a></li>
<li class="chapter" data-level="9.8.4" data-path="cell_type.html"><a href="cell_type.html#clustering-by-branch-detection-along-pseudotime-optional"><i class="fa fa-check"></i><b>9.8.4</b> Clustering by branch detection along pseudotime (optional)</a></li>
<li class="chapter" data-level="9.8.5" data-path="cell_type.html"><a href="cell_type.html#save-data"><i class="fa fa-check"></i><b>9.8.5</b> Save data</a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="cell_type.html"><a href="cell_type.html#visualize-clustering-results"><i class="fa fa-check"></i><b>9.9</b> Visualize clustering results</a></li>
<li class="chapter" data-level="9.10" data-path="cell_type.html"><a href="cell_type.html#find-significant-signs"><i class="fa fa-check"></i><b>9.10</b> Find significant signs</a>
<ul>
<li class="chapter" data-level="9.10.1" data-path="cell_type.html"><a href="cell_type.html#compute-separation-indices"><i class="fa fa-check"></i><b>9.10.1</b> Compute separation indices</a></li>
<li class="chapter" data-level="9.10.2" data-path="cell_type.html"><a href="cell_type.html#investigate-sign-scores"><i class="fa fa-check"></i><b>9.10.2</b> Investigate sign scores</a></li>
</ul></li>
<li class="chapter" data-level="9.11" data-path="cell_type.html"><a href="cell_type.html#investigate-batch-effects"><i class="fa fa-check"></i><b>9.11</b> Investigate batch effects</a></li>
<li class="chapter" data-level="9.12" data-path="cell_type.html"><a href="cell_type.html#spatial-mapping-by-spaniel"><i class="fa fa-check"></i><b>9.12</b> Spatial mapping by Spaniel</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biological_function.html"><a href="biological_function.html"><i class="fa fa-check"></i><b>10</b> Inferring biological functions (optional)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biological_function.html"><a href="biological_function.html#add-database-to-asurat-objects-1"><i class="fa fa-check"></i><b>10.1</b> Add database to ASURAT objects</a></li>
<li class="chapter" data-level="10.2" data-path="biological_function.html"><a href="biological_function.html#quick-qc-2"><i class="fa fa-check"></i><b>10.2</b> Quick QC</a></li>
<li class="chapter" data-level="10.3" data-path="biological_function.html"><a href="biological_function.html#correlation-graph-based-decomposition-1"><i class="fa fa-check"></i><b>10.3</b> Correlation graph-based decomposition</a></li>
<li class="chapter" data-level="10.4" data-path="biological_function.html"><a href="biological_function.html#creating-systems-of-signs-1"><i class="fa fa-check"></i><b>10.4</b> Creating systems of signs</a></li>
<li class="chapter" data-level="10.5" data-path="biological_function.html"><a href="biological_function.html#removing-redundant-signs-1"><i class="fa fa-check"></i><b>10.5</b> Removing redundant signs</a></li>
<li class="chapter" data-level="10.6" data-path="biological_function.html"><a href="biological_function.html#creating-sign-by-sample-matrices-1"><i class="fa fa-check"></i><b>10.6</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="10.7" data-path="biological_function.html"><a href="biological_function.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>10.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="biological_function.html"><a href="biological_function.html#principal-component-analysis-optional-1"><i class="fa fa-check"></i><b>10.7.1</b> Principal component analysis (optional)</a></li>
<li class="chapter" data-level="10.7.2" data-path="biological_function.html"><a href="biological_function.html#t-distributed-stochastic-neighbor-embedding-optional-1"><i class="fa fa-check"></i><b>10.7.2</b> t-distributed stochastic neighbor embedding (optional)</a></li>
<li class="chapter" data-level="10.7.3" data-path="biological_function.html"><a href="biological_function.html#uniform-manifold-approximation-and-projection-optional-1"><i class="fa fa-check"></i><b>10.7.3</b> Uniform manifold approximation and projection (optional)</a></li>
<li class="chapter" data-level="10.7.4" data-path="biological_function.html"><a href="biological_function.html#diffusion-map-optional-1"><i class="fa fa-check"></i><b>10.7.4</b> Diffusion map (optional)</a></li>
<li class="chapter" data-level="10.7.5" data-path="biological_function.html"><a href="biological_function.html#visualization-1"><i class="fa fa-check"></i><b>10.7.5</b> Visualization</a></li>
<li class="chapter" data-level="10.7.6" data-path="biological_function.html"><a href="biological_function.html#save-results.-1"><i class="fa fa-check"></i><b>10.7.6</b> Save results.</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="biological_function.html"><a href="biological_function.html#cell-clustering-1"><i class="fa fa-check"></i><b>10.8</b> Cell clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="biological_function.html"><a href="biological_function.html#partitioning-around-medoids-pam-optional-1"><i class="fa fa-check"></i><b>10.8.1</b> Partitioning around medoids (PAM) (optional)</a></li>
<li class="chapter" data-level="10.8.2" data-path="biological_function.html"><a href="biological_function.html#hierarchical-clustering-and-cutting-tree-optional-1"><i class="fa fa-check"></i><b>10.8.2</b> Hierarchical clustering and cutting tree (optional)</a></li>
<li class="chapter" data-level="10.8.3" data-path="biological_function.html"><a href="biological_function.html#community-detection-algorithm-with-k-nearest-neighbor-knn-graph-optional-1"><i class="fa fa-check"></i><b>10.8.3</b> Community detection algorithm with k-nearest neighbor (KNN) graph (optional)</a></li>
<li class="chapter" data-level="10.8.4" data-path="biological_function.html"><a href="biological_function.html#clustering-by-branch-detection-along-pseudotime-optional-1"><i class="fa fa-check"></i><b>10.8.4</b> Clustering by branch detection along pseudotime (optional)</a></li>
<li class="chapter" data-level="10.8.5" data-path="biological_function.html"><a href="biological_function.html#save-data-1"><i class="fa fa-check"></i><b>10.8.5</b> Save data</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="biological_function.html"><a href="biological_function.html#visualize-clustering-results-1"><i class="fa fa-check"></i><b>10.9</b> Visualize clustering results</a></li>
<li class="chapter" data-level="10.10" data-path="biological_function.html"><a href="biological_function.html#find-significant-signs-1"><i class="fa fa-check"></i><b>10.10</b> Find significant signs</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="biological_function.html"><a href="biological_function.html#compute-separation-indices-1"><i class="fa fa-check"></i><b>10.10.1</b> Compute separation indices</a></li>
<li class="chapter" data-level="10.10.2" data-path="biological_function.html"><a href="biological_function.html#investigate-sign-scores-1"><i class="fa fa-check"></i><b>10.10.2</b> Investigate sign scores</a></li>
</ul></li>
<li class="chapter" data-level="10.11" data-path="biological_function.html"><a href="biological_function.html#investigate-batch-effects-1"><i class="fa fa-check"></i><b>10.11</b> Investigate batch effects</a></li>
<li class="chapter" data-level="10.12" data-path="biological_function.html"><a href="biological_function.html#spatial-mapping-by-spaniel-1"><i class="fa fa-check"></i><b>10.12</b> Spatial mapping by Spaniel</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pathway.html"><a href="pathway.html"><i class="fa fa-check"></i><b>11</b> Inferring pathway activities (optional)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="pathway.html"><a href="pathway.html#add-database-to-asurat-objects-2"><i class="fa fa-check"></i><b>11.1</b> Add database to ASURAT objects</a></li>
<li class="chapter" data-level="11.2" data-path="pathway.html"><a href="pathway.html#quick-qc-3"><i class="fa fa-check"></i><b>11.2</b> Quick QC</a></li>
<li class="chapter" data-level="11.3" data-path="pathway.html"><a href="pathway.html#correlation-graph-based-decomposition-2"><i class="fa fa-check"></i><b>11.3</b> Correlation graph-based decomposition</a></li>
<li class="chapter" data-level="11.4" data-path="pathway.html"><a href="pathway.html#creating-systems-of-signs-2"><i class="fa fa-check"></i><b>11.4</b> Creating systems of signs</a></li>
<li class="chapter" data-level="11.5" data-path="pathway.html"><a href="pathway.html#removing-redundant-signs-2"><i class="fa fa-check"></i><b>11.5</b> Removing redundant signs</a></li>
<li class="chapter" data-level="11.6" data-path="pathway.html"><a href="pathway.html#creating-sign-by-sample-matrices-2"><i class="fa fa-check"></i><b>11.6</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="11.7" data-path="pathway.html"><a href="pathway.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>11.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="pathway.html"><a href="pathway.html#principal-component-analysis-optional-2"><i class="fa fa-check"></i><b>11.7.1</b> Principal component analysis (optional)</a></li>
<li class="chapter" data-level="11.7.2" data-path="pathway.html"><a href="pathway.html#t-distributed-stochastic-neighbor-embedding-optional-2"><i class="fa fa-check"></i><b>11.7.2</b> t-distributed stochastic neighbor embedding (optional)</a></li>
<li class="chapter" data-level="11.7.3" data-path="pathway.html"><a href="pathway.html#uniform-manifold-approximation-and-projection-optional-2"><i class="fa fa-check"></i><b>11.7.3</b> Uniform manifold approximation and projection (optional)</a></li>
<li class="chapter" data-level="11.7.4" data-path="pathway.html"><a href="pathway.html#diffusion-map-optional-2"><i class="fa fa-check"></i><b>11.7.4</b> Diffusion map (optional)</a></li>
<li class="chapter" data-level="11.7.5" data-path="pathway.html"><a href="pathway.html#visualization-2"><i class="fa fa-check"></i><b>11.7.5</b> Visualization</a></li>
<li class="chapter" data-level="11.7.6" data-path="pathway.html"><a href="pathway.html#save-results.-2"><i class="fa fa-check"></i><b>11.7.6</b> Save results.</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="pathway.html"><a href="pathway.html#cell-clustering-2"><i class="fa fa-check"></i><b>11.8</b> Cell clustering</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="pathway.html"><a href="pathway.html#partitioning-around-medoids-pam-optional-2"><i class="fa fa-check"></i><b>11.8.1</b> Partitioning around medoids (PAM) (optional)</a></li>
<li class="chapter" data-level="11.8.2" data-path="pathway.html"><a href="pathway.html#hierarchical-clustering-and-cutting-tree-optional-2"><i class="fa fa-check"></i><b>11.8.2</b> Hierarchical clustering and cutting tree (optional)</a></li>
<li class="chapter" data-level="11.8.3" data-path="pathway.html"><a href="pathway.html#community-detection-algorithm-with-k-nearest-neighbor-knn-graph-optional-2"><i class="fa fa-check"></i><b>11.8.3</b> Community detection algorithm with k-nearest neighbor (KNN) graph (optional)</a></li>
<li class="chapter" data-level="11.8.4" data-path="pathway.html"><a href="pathway.html#clustering-by-branch-detection-along-pseudotime-optional-2"><i class="fa fa-check"></i><b>11.8.4</b> Clustering by branch detection along pseudotime (optional)</a></li>
<li class="chapter" data-level="11.8.5" data-path="pathway.html"><a href="pathway.html#save-data-2"><i class="fa fa-check"></i><b>11.8.5</b> Save data</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="pathway.html"><a href="pathway.html#visualize-clustering-results-2"><i class="fa fa-check"></i><b>11.9</b> Visualize clustering results</a></li>
<li class="chapter" data-level="11.10" data-path="pathway.html"><a href="pathway.html#find-significant-signs-2"><i class="fa fa-check"></i><b>11.10</b> Find significant signs</a>
<ul>
<li class="chapter" data-level="11.10.1" data-path="pathway.html"><a href="pathway.html#compute-separation-indices-2"><i class="fa fa-check"></i><b>11.10.1</b> Compute separation indices</a></li>
<li class="chapter" data-level="11.10.2" data-path="pathway.html"><a href="pathway.html#investigate-sign-scores-2"><i class="fa fa-check"></i><b>11.10.2</b> Investigate sign scores</a></li>
</ul></li>
<li class="chapter" data-level="11.11" data-path="pathway.html"><a href="pathway.html#investigate-batch-effects-2"><i class="fa fa-check"></i><b>11.11</b> Investigate batch effects</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="multi_layered.html"><a href="multi_layered.html"><i class="fa fa-check"></i><b>12</b> Multi-layered analysis across all the signs</a>
<ul>
<li class="chapter" data-level="12.1" data-path="multi_layered.html"><a href="multi_layered.html#concatenate-asurat-objects"><i class="fa fa-check"></i><b>12.1</b> Concatenate ASURAT objects</a></li>
<li class="chapter" data-level="12.2" data-path="multi_layered.html"><a href="multi_layered.html#find-significant-signs-and-genes-across-different-data-types"><i class="fa fa-check"></i><b>12.2</b> Find significant signs and genes across different data types</a></li>
<li class="chapter" data-level="12.3" data-path="multi_layered.html"><a href="multi_layered.html#spatial-mapping-by-spaniel-2"><i class="fa fa-check"></i><b>12.3</b> Spatial mapping by Spaniel</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><i class="fa fa-check"></i><b>13</b> Appendix A: analyzing scRNA-seq data by existing method I</a>
<ul>
<li class="chapter" data-level="13.1" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#preprocessing"><i class="fa fa-check"></i><b>13.1</b> Preprocessing</a></li>
<li class="chapter" data-level="13.2" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#cluster-cells"><i class="fa fa-check"></i><b>13.2</b> Cluster cells</a></li>
<li class="chapter" data-level="13.3" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#visualize-data-in-low-dimensional-space"><i class="fa fa-check"></i><b>13.3</b> Visualize data in low dimensional space</a></li>
<li class="chapter" data-level="13.4" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#find-differentially-expressed-genes"><i class="fa fa-check"></i><b>13.4</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="13.5" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#save-results-1"><i class="fa fa-check"></i><b>13.5</b> Save results</a></li>
<li class="chapter" data-level="13.6" data-path="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html"><a href="appendix-a-analyzing-scrna-seq-data-by-existing-method-i.html#infer-cell-types-for-pbmc-datasets"><i class="fa fa-check"></i><b>13.6</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><i class="fa fa-check"></i><b>14</b> Appendix B: analyzing scRNA-seq data by existing method II</a>
<ul>
<li class="chapter" data-level="14.1" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#preprocessing-1"><i class="fa fa-check"></i><b>14.1</b> Preprocessing</a></li>
<li class="chapter" data-level="14.2" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cluster-cells-1"><i class="fa fa-check"></i><b>14.2</b> Cluster cells</a></li>
<li class="chapter" data-level="14.3" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#visualize-data-in-low-dimensional-space-1"><i class="fa fa-check"></i><b>14.3</b> Visualize data in low dimensional space</a></li>
<li class="chapter" data-level="14.4" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#cell-cycle-scoring"><i class="fa fa-check"></i><b>14.4</b> Cell cycle scoring</a></li>
<li class="chapter" data-level="14.5" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#find-differentially-expressed-genes-1"><i class="fa fa-check"></i><b>14.5</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="14.6" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#enrichment-analysis"><i class="fa fa-check"></i><b>14.6</b> Enrichment analysis</a></li>
<li class="chapter" data-level="14.7" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#save-results-2"><i class="fa fa-check"></i><b>14.7</b> Save results</a></li>
<li class="chapter" data-level="14.8" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#infer-cell-types-for-pbmc-datasets-1"><i class="fa fa-check"></i><b>14.8</b> Infer cell types for PBMC datasets</a></li>
<li class="chapter" data-level="14.9" data-path="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html"><a href="appendix-b-analyzing-scrna-seq-data-by-existing-method-ii.html#infer-cell-types-for-pbmc-datasets-by-seurat-and-sccatch"><i class="fa fa-check"></i><b>14.9</b> Infer cell types for PBMC datasets by Seurat and scCATCH</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><i class="fa fa-check"></i><b>15</b> Appendix C: analyzing scRNA-seq data by existing method III</a>
<ul>
<li class="chapter" data-level="15.1" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#preprocessing-2"><i class="fa fa-check"></i><b>15.1</b> Preprocessing</a></li>
<li class="chapter" data-level="15.2" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#cluster-cells-2"><i class="fa fa-check"></i><b>15.2</b> Cluster cells</a></li>
<li class="chapter" data-level="15.3" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#visualize-data-in-low-dimensional-space-2"><i class="fa fa-check"></i><b>15.3</b> Visualize data in low dimensional space</a></li>
<li class="chapter" data-level="15.4" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#find-differentially-expressed-genes-2"><i class="fa fa-check"></i><b>15.4</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="15.5" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#save-results-3"><i class="fa fa-check"></i><b>15.5</b> Save results</a></li>
<li class="chapter" data-level="15.6" data-path="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html"><a href="appendix-c-analyzing-scrna-seq-data-by-existing-method-iii.html#infer-cell-types-for-pbmc-datasets-2"><i class="fa fa-check"></i><b>15.6</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><i class="fa fa-check"></i><b>16</b> Appendix D: analyzing scRNA-seq data by existing method IV</a>
<ul>
<li class="chapter" data-level="16.1" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#preprocessing-3"><i class="fa fa-check"></i><b>16.1</b> Preprocessing</a></li>
<li class="chapter" data-level="16.2" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#cluster-cells-3"><i class="fa fa-check"></i><b>16.2</b> Cluster cells</a></li>
<li class="chapter" data-level="16.3" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#find-differentially-expressed-genes-3"><i class="fa fa-check"></i><b>16.3</b> Find differentially expressed genes</a></li>
<li class="chapter" data-level="16.4" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#save-results-4"><i class="fa fa-check"></i><b>16.4</b> Save results</a></li>
<li class="chapter" data-level="16.5" data-path="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html"><a href="appendix-d-analyzing-scrna-seq-data-by-existing-method-iv.html#infer-cell-types-for-pbmc-datasets-3"><i class="fa fa-check"></i><b>16.5</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="appendix-e-analyzing-scrna-seq-data-by-existing-method-v.html"><a href="appendix-e-analyzing-scrna-seq-data-by-existing-method-v.html"><i class="fa fa-check"></i><b>17</b> Appendix E: analyzing scRNA-seq data by existing method V</a>
<ul>
<li class="chapter" data-level="17.1" data-path="appendix-e-analyzing-scrna-seq-data-by-existing-method-v.html"><a href="appendix-e-analyzing-scrna-seq-data-by-existing-method-v.html#infer-cell-types-for-pbmc-datasets-4"><i class="fa fa-check"></i><b>17.1</b> Infer cell types for PBMC datasets</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html"><i class="fa fa-check"></i><b>18</b> Appendix F: creating figures</a>
<ul>
<li class="chapter" data-level="18.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#pbmcs-datasets"><i class="fa fa-check"></i><b>18.1</b> PBMCs datasets</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#multi-layered-analysis"><i class="fa fa-check"></i><b>18.1.1</b> Multi-layered analysis</a></li>
<li class="chapter" data-level="18.1.2" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#infer-population-ratios"><i class="fa fa-check"></i><b>18.1.2</b> Infer population ratios</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#lung-cancer-dataset"><i class="fa fa-check"></i><b>18.2</b> Lung cancer dataset</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#heatmaps-of-marker-gene-expressions"><i class="fa fa-check"></i><b>18.2.1</b> Heatmaps of marker gene expressions</a></li>
<li class="chapter" data-level="18.2.2" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#multi-layered-analysis-1"><i class="fa fa-check"></i><b>18.2.2</b> Multi-layered analysis</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#pancreatic-ductal-adenocarcinoma-pdac-datasets"><i class="fa fa-check"></i><b>18.3</b> Pancreatic ductal adenocarcinoma (PDAC) datasets</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="appendix-f-creating-figures.html"><a href="appendix-f-creating-figures.html#clustering-pdac-spatial-transcriptomes"><i class="fa fa-check"></i><b>18.3.1</b> Clustering PDAC spatial transcriptomes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to ASURAT (version 0.0.0.9001)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="collectdb" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Collecting databases (optional)</h1>
<p>In this section, the following databases (DBs) are downloaded and properly
formatted.
Since all of the results in this section have already been prepared for human
and mouse datasets in December 2020 (see <code>data/2020_001_databases/</code>),
users may be able to skip this section.</p>
<ol style="list-style-type: decimal">
<li>DB for Disease Ontology (DO) for human disease</li>
<li>DB for Cell Ontology (CO)</li>
<li>DB for Gene Ontology (GO)</li>
<li>DB for Kyoto Encyclopedia of Genes and Genomes (KEGG)</li>
<li>DB for Reactome</li>
</ol>
<div id="disease-ontology-db" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Disease Ontology DB</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="collectdb.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_DiseaseOntology.R&quot;</span>)</span></code></pre></div>
<p>The following functions <code>collect_DO()</code>, <code>format_DO()</code>, and <code>compute_IC_DO()</code>
(i) downloads a database for Disease Ontology,
(ii) formats the results, and
(iii) prepares a list of information content (IC) (Mistry and Pavlidis,
BMC Bioinformatics, 2008), respectively.
IC value represents the depth of the biological term in the ontology,
which will be later used for computing similarities between terms.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="collectdb.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-2"><a href="collectdb.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note again that the following result has already been prepared for human and</span></span>
<span id="cb2-3"><a href="collectdb.html#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mouse datasets in December 2020 (see `data/2020_001_databases/`).</span></span>
<span id="cb2-4"><a href="collectdb.html#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-5"><a href="collectdb.html#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DO.db)</span>
<span id="cb2-6"><a href="collectdb.html#cb2-6" aria-hidden="true" tabindex="-1"></a>dict_DO <span class="ot">&lt;-</span> <span class="fu">collect_DO</span>()</span>
<span id="cb2-7"><a href="collectdb.html#cb2-7" aria-hidden="true" tabindex="-1"></a>tidy_DO <span class="ot">&lt;-</span> <span class="fu">format_DO</span>(<span class="at">dict =</span> dict_DO)</span>
<span id="cb2-8"><a href="collectdb.html#cb2-8" aria-hidden="true" tabindex="-1"></a>IC_DO <span class="ot">&lt;-</span> <span class="fu">compute_IC_DO</span>(<span class="at">tidy =</span> tidy_DO)</span>
<span id="cb2-9"><a href="collectdb.html#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-10"><a href="collectdb.html#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the result.</span></span>
<span id="cb2-11"><a href="collectdb.html#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-12"><a href="collectdb.html#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dict_DO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_dict_DO_human.rds&quot;</span>)</span>
<span id="cb2-13"><a href="collectdb.html#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tidy_DO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_tidy_DO_human.rds&quot;</span>)</span>
<span id="cb2-14"><a href="collectdb.html#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(IC_DO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_IC_DO_human.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="cell-ontology-db" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Cell Ontology DB</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="collectdb.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_CellOntology.R&quot;</span>)</span></code></pre></div>
<p>The following functions <code>collect_CO()</code>, <code>format_CO()</code>, <code>make_treeTable_CO()</code>,
and <code>compute_IC_CO()</code>
(i) downloads a database for Disease Ontology,
(ii) formats the results, and
(iii) prepares a list of information content (IC) (Mistry and Pavlidis,
BMC Bioinformatics, 2008), respectively.</p>
<p><strong>Tips:</strong>
As of December 2020, Cell Ontology database might not be complete enough
for some biological contexts.
For example, the well-known marker genes for pancreatic beta cell,
Ins1 and Ins2, are not registered in the gene slot of <code>type B pancreatic cell</code>
with CO ID <code>CL:0000169</code>.
Be careful of using IC values of CO, which may not be always useful for
computing similarities between biological terms.
Nevertheless, we computed the IC values as in the future CO database might be
better annotated.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="collectdb.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb4-2"><a href="collectdb.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note again that the following result has already been prepared for human and</span></span>
<span id="cb4-3"><a href="collectdb.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mouse datasets in December 2020 (see `data/2020_001_databases/`).</span></span>
<span id="cb4-4"><a href="collectdb.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb4-5"><a href="collectdb.html#cb4-5" aria-hidden="true" tabindex="-1"></a>dict_CO <span class="ot">&lt;-</span> <span class="fu">collect_CO</span>(<span class="at">orgdb =</span> org.Hs.eg.db)</span>
<span id="cb4-6"><a href="collectdb.html#cb4-6" aria-hidden="true" tabindex="-1"></a>tidy_CO <span class="ot">&lt;-</span> <span class="fu">format_CO</span>(<span class="at">dict =</span> dict_CO)</span>
<span id="cb4-7"><a href="collectdb.html#cb4-7" aria-hidden="true" tabindex="-1"></a>treeTable_CO <span class="ot">&lt;-</span> <span class="fu">make_treeTable_CO</span>(<span class="at">tidy =</span> tidy_CO)</span>
<span id="cb4-8"><a href="collectdb.html#cb4-8" aria-hidden="true" tabindex="-1"></a>IC_CO <span class="ot">&lt;-</span> <span class="fu">compute_IC_CO</span>(<span class="at">tidy =</span> tidy_CO, <span class="at">treeTable =</span> treeTable_CO)</span>
<span id="cb4-9"><a href="collectdb.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb4-10"><a href="collectdb.html#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the result.</span></span>
<span id="cb4-11"><a href="collectdb.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb4-12"><a href="collectdb.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dict_CO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_dict_CO_human.rds&quot;</span>)</span>
<span id="cb4-13"><a href="collectdb.html#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tidy_CO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_tidy_CO_human.rds&quot;</span>)</span>
<span id="cb4-14"><a href="collectdb.html#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(treeTable_CO,</span>
<span id="cb4-15"><a href="collectdb.html#cb4-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_treeTable_CO_human.rds&quot;</span>)</span>
<span id="cb4-16"><a href="collectdb.html#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(IC_CO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_IC_CO_human.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="gene-ontology-db" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Gene Ontology DB</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="collectdb.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_GeneOntology.R&quot;</span>)</span></code></pre></div>
<p>The following functions <code>collect_GO()</code>, <code>format_GO()</code>, and <code>compute_IC_GO()</code>
(i) downloads a database for Gene Ontology,
(ii) formats the results, and
(iii) prepares a list of information content (IC), respectively.
The arguments of <code>collect_GO()</code> are <code>orgdb</code>
(annotation data package such as <code>org.Hs.eg.db</code> and <code>org.Mm.eg.db</code>) and
<code>all_geneIDs</code> (all Entrez gene IDs of a specific organism).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="collectdb.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb6-2"><a href="collectdb.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note again that the following result has already been prepared for human and</span></span>
<span id="cb6-3"><a href="collectdb.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mouse datasets in December 2020 (see `data/2020_001_databases/`).</span></span>
<span id="cb6-4"><a href="collectdb.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb6-5"><a href="collectdb.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb6-6"><a href="collectdb.html#cb6-6" aria-hidden="true" tabindex="-1"></a>all_geneIDs <span class="ot">&lt;-</span> <span class="fu">as.list</span>(org.Hs.egGO)</span>
<span id="cb6-7"><a href="collectdb.html#cb6-7" aria-hidden="true" tabindex="-1"></a>all_geneIDs <span class="ot">&lt;-</span> <span class="fu">names</span>(all_geneIDs[<span class="sc">!</span><span class="fu">is.na</span>(all_geneIDs)])</span>
<span id="cb6-8"><a href="collectdb.html#cb6-8" aria-hidden="true" tabindex="-1"></a>dict_GO <span class="ot">&lt;-</span> <span class="fu">collect_GO</span>(<span class="at">orgdb =</span> org.Hs.eg.db, <span class="at">all_geneIDs =</span> all_geneIDs)</span>
<span id="cb6-9"><a href="collectdb.html#cb6-9" aria-hidden="true" tabindex="-1"></a>tidy_GO <span class="ot">&lt;-</span> <span class="fu">format_GO</span>(<span class="at">dict =</span> dict_GO)</span>
<span id="cb6-10"><a href="collectdb.html#cb6-10" aria-hidden="true" tabindex="-1"></a>IC_GO <span class="ot">&lt;-</span> <span class="fu">compute_IC_GO</span>(<span class="at">tidy =</span> tidy_GO)</span>
<span id="cb6-11"><a href="collectdb.html#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb6-12"><a href="collectdb.html#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the result.</span></span>
<span id="cb6-13"><a href="collectdb.html#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb6-14"><a href="collectdb.html#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dict_DO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_dict_GO_human.rds&quot;</span>)</span>
<span id="cb6-15"><a href="collectdb.html#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tidy_DO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_tidy_GO_human.rds&quot;</span>)</span>
<span id="cb6-16"><a href="collectdb.html#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(IC_DO, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_IC_GO_human.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="kegg" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> KEGG</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="collectdb.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;R/function_KEGG.R&quot;</span>)</span></code></pre></div>
<p>The following functions <code>collect_KEGG()</code> and <code>format_KEGG()</code>
(i) downloads a database for KEGG and (ii) formats the results, respectively.
In the current version, KEGG PATHWAY (<code>"pathway"</code>) and MODULE (<code>"module"</code>)
can be input into <code>collect_KEGG()</code> (cf.<code>KEGGREST</code> function <code>listDatabases()</code>).
The arguments of <code>collect_KEGG()</code> are <code>organism</code> (such as Homo sapiens <code>"hsa"</code>,
obeying the naming rule of <a href="http://rest.kegg.jp/list/organism">KEGG</a>) and
<code>category_names</code> (such as <code>"pathway"</code> and <code>"module"</code>).
This function requires an internet connection.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="collectdb.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb8-2"><a href="collectdb.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note again that the following result has already been prepared for human and</span></span>
<span id="cb8-3"><a href="collectdb.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mouse datasets in December 2020 (see `data/2020_001_databases/`).</span></span>
<span id="cb8-4"><a href="collectdb.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb8-5"><a href="collectdb.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that `collect_KEGG` uses `KEGGREST` function `keggGet` with internet</span></span>
<span id="cb8-6"><a href="collectdb.html#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># connection, which may produce both successful and unsuccessful results.</span></span>
<span id="cb8-7"><a href="collectdb.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb8-8"><a href="collectdb.html#cb8-8" aria-hidden="true" tabindex="-1"></a>dict_KEGG_raw <span class="ot">&lt;-</span> <span class="fu">collect_KEGG</span>(<span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>, <span class="at">category_names =</span> <span class="fu">c</span>(<span class="st">&quot;pathway&quot;</span>))</span>
<span id="cb8-9"><a href="collectdb.html#cb8-9" aria-hidden="true" tabindex="-1"></a>dict_KEGG <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">pathway =</span> dict_KEGG_raw[[<span class="st">&quot;pathway&quot;</span>]][[<span class="st">&quot;success&quot;</span>]])</span>
<span id="cb8-10"><a href="collectdb.html#cb8-10" aria-hidden="true" tabindex="-1"></a>tidy_KEGG <span class="ot">&lt;-</span> <span class="fu">format_KEGG</span>(<span class="at">dict =</span> dict_KEGG)</span>
<span id="cb8-11"><a href="collectdb.html#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb8-12"><a href="collectdb.html#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the result.</span></span>
<span id="cb8-13"><a href="collectdb.html#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb8-14"><a href="collectdb.html#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dict_KEGG_raw,</span>
<span id="cb8-15"><a href="collectdb.html#cb8-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_dict_KEGG_raw_human.rds&quot;</span>)</span>
<span id="cb8-16"><a href="collectdb.html#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dict_KEGG,</span>
<span id="cb8-17"><a href="collectdb.html#cb8-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_dict_KEGG_human.rds&quot;</span>)</span>
<span id="cb8-18"><a href="collectdb.html#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tidy_KEGG,</span>
<span id="cb8-19"><a href="collectdb.html#cb8-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/YYYYMMDD_tidy_KEGG_human.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="custom-built-db-for-our-analysis-of-pdac-data" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Custom-built DB for our analysis of PDAC data</h2>
<p>To facilitate our analysis of PDAC data (Moncada et al., Nat Biotechnol, 2020),
we downloaded some of the databases and combined them with DO and CO,
which we refer as <code>CB</code> with category name <code>cell</code>
(see <code>data/2020_001_databases/20210722_tidy_CB_human.rds</code>).</p>
<p>Load Disease Ontology (DO) and Cell Ontology (CO) databases and
convert gene symbols into entrez IDs using <code>org.Hs.eg.db</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="collectdb.html#cb9-1" aria-hidden="true" tabindex="-1"></a>tidy_DO <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/20201213_tidy_DO_human.rds&quot;</span>)</span>
<span id="cb9-2"><a href="collectdb.html#cb9-2" aria-hidden="true" tabindex="-1"></a>dict_CO <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/20201213_dict_CO_human.rds&quot;</span>)</span>
<span id="cb9-3"><a href="collectdb.html#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="collectdb.html#cb9-4" aria-hidden="true" tabindex="-1"></a>dictionary <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db,</span>
<span id="cb9-5"><a href="collectdb.html#cb9-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">key =</span> dict_CO<span class="sc">$</span>Symbol,</span>
<span id="cb9-6"><a href="collectdb.html#cb9-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;ENTREZID&quot;</span>),</span>
<span id="cb9-7"><a href="collectdb.html#cb9-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb9-8"><a href="collectdb.html#cb9-8" aria-hidden="true" tabindex="-1"></a>dictionary <span class="ot">&lt;-</span> dictionary[<span class="sc">!</span><span class="fu">duplicated</span>(dictionary<span class="sc">$</span>SYMBOL), ]</span>
<span id="cb9-9"><a href="collectdb.html#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(dict_CO)){</span>
<span id="cb9-10"><a href="collectdb.html#cb9-10" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> dict_CO<span class="sc">$</span>Symbol[i]</span>
<span id="cb9-11"><a href="collectdb.html#cb9-11" aria-hidden="true" tabindex="-1"></a>  dict_CO<span class="sc">$</span>Symbol[i] <span class="ot">&lt;-</span> dictionary[<span class="fu">which</span>(dictionary<span class="sc">$</span>SYMBOL <span class="sc">==</span> gene),]<span class="sc">$</span>ENTREZID</span>
<span id="cb9-12"><a href="collectdb.html#cb9-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>The following libraries were installed on 11th July, 2021.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="collectdb.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSEABase)        <span class="co"># For using clustermole</span></span>
<span id="cb10-2"><a href="collectdb.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)            <span class="co"># For using clustermole</span></span>
<span id="cb10-3"><a href="collectdb.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(singscore)       <span class="co"># For using clustermole</span></span>
<span id="cb10-4"><a href="collectdb.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clustermole)     <span class="co"># For using clustermole</span></span></code></pre></div>
<p>Load the database and check the cell type source databases.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="collectdb.html#cb11-1" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">clustermole_markers</span>()</span>
<span id="cb11-2"><a href="collectdb.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">unique</span>(markers<span class="sc">$</span>db))</span></code></pre></div>
<pre><code>[1] &quot;ARCHS4&quot;     &quot;CellMarker&quot; &quot;MSigDB&quot;     &quot;PanglaoDB&quot;  &quot;SaVanT&quot;     &quot;TISSUES&quot;   
[7] &quot;xCell&quot; </code></pre>
<p>Select <code>species</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="collectdb.html#cb13-1" aria-hidden="true" tabindex="-1"></a>markers_red <span class="ot">&lt;-</span> markers[<span class="fu">which</span>(markers<span class="sc">$</span>species <span class="sc">==</span> <span class="st">&quot;Human&quot;</span>),]</span></code></pre></div>
<p>Select <code>db</code> (database).</p>
<p><strong>Tips:</strong>
Since we are interested in prostate ductal adenocarcinoma (PDAC) and
other pancreas related cells such as ductal cell, acinar cell, etc,
we carefully select the following DBs.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="collectdb.html#cb14-1" aria-hidden="true" tabindex="-1"></a>markers_red <span class="ot">&lt;-</span> markers_red[<span class="fu">which</span>(markers_red<span class="sc">$</span>db <span class="sc">==</span> <span class="st">&quot;CellMarker&quot;</span> <span class="sc">|</span></span>
<span id="cb14-2"><a href="collectdb.html#cb14-2" aria-hidden="true" tabindex="-1"></a>                                   markers_red<span class="sc">$</span>db <span class="sc">==</span> <span class="st">&quot;MSigDB&quot;</span> <span class="sc">|</span></span>
<span id="cb14-3"><a href="collectdb.html#cb14-3" aria-hidden="true" tabindex="-1"></a>                                   markers_red<span class="sc">$</span>db <span class="sc">==</span> <span class="st">&quot;PanglaoDB&quot;</span>),]</span></code></pre></div>
<p>Change gene symbols into entrez IDs using <code>org.Hs.eg.db</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="collectdb.html#cb15-1" aria-hidden="true" tabindex="-1"></a>dictionary <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db,</span>
<span id="cb15-2"><a href="collectdb.html#cb15-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">key =</span> markers_red<span class="sc">$</span>gene,</span>
<span id="cb15-3"><a href="collectdb.html#cb15-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;ENTREZID&quot;</span>),</span>
<span id="cb15-4"><a href="collectdb.html#cb15-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb15-5"><a href="collectdb.html#cb15-5" aria-hidden="true" tabindex="-1"></a>dictionary <span class="ot">&lt;-</span> dictionary[<span class="sc">!</span><span class="fu">duplicated</span>(dictionary<span class="sc">$</span>SYMBOL), ]</span>
<span id="cb15-6"><a href="collectdb.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(markers_red)){</span>
<span id="cb15-7"><a href="collectdb.html#cb15-7" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> markers_red<span class="sc">$</span>gene[i]</span>
<span id="cb15-8"><a href="collectdb.html#cb15-8" aria-hidden="true" tabindex="-1"></a>  markers_red<span class="sc">$</span>gene[i] <span class="ot">&lt;-</span> dictionary[<span class="fu">which</span>(dictionary<span class="sc">$</span>SYMBOL <span class="sc">==</span> gene),]<span class="sc">$</span>ENTREZID</span>
<span id="cb15-9"><a href="collectdb.html#cb15-9" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Format the database.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="collectdb.html#cb16-1" aria-hidden="true" tabindex="-1"></a>markers_list <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="at">x =</span> markers_red<span class="sc">$</span>gene, <span class="at">f =</span> markers_red<span class="sc">$</span>celltype)</span>
<span id="cb16-2"><a href="collectdb.html#cb16-2" aria-hidden="true" tabindex="-1"></a>nlist <span class="ot">&lt;-</span> <span class="fu">length</span>(markers_list)</span>
<span id="cb16-3"><a href="collectdb.html#cb16-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol =</span> <span class="dv">5</span>, <span class="at">nrow =</span> nlist, <span class="at">dimnames =</span> <span class="fu">list</span>(<span class="cn">NULL</span>, <span class="fu">c</span>(</span>
<span id="cb16-4"><a href="collectdb.html#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;ID&quot;</span>, <span class="st">&quot;Description&quot;</span>, <span class="st">&quot;Count&quot;</span>, <span class="st">&quot;Gene&quot;</span>, <span class="st">&quot;GeneID&quot;</span>))))</span>
<span id="cb16-5"><a href="collectdb.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nlist){</span>
<span id="cb16-6"><a href="collectdb.html#cb16-6" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> markers_list[[i]]</span>
<span id="cb16-7"><a href="collectdb.html#cb16-7" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>ID[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;CBID:&quot;</span>, i, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb16-8"><a href="collectdb.html#cb16-8" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>Description[i] <span class="ot">&lt;-</span> <span class="fu">names</span>(markers_list)[i]</span>
<span id="cb16-9"><a href="collectdb.html#cb16-9" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>Count[i] <span class="ot">&lt;-</span> <span class="fu">length</span>(markers_list[[i]])</span>
<span id="cb16-10"><a href="collectdb.html#cb16-10" aria-hidden="true" tabindex="-1"></a>  res<span class="sc">$</span>GeneID[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>(markers_list[[i]], <span class="at">collapse =</span> <span class="st">&quot;/&quot;</span>)</span>
<span id="cb16-11"><a href="collectdb.html#cb16-11" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Create a structured data frame by combining <code>res</code> with <code>tidy_DO</code> and <code>dict_CO</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="collectdb.html#cb17-1" aria-hidden="true" tabindex="-1"></a>map <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">data.frame</span>(</span>
<span id="cb17-2"><a href="collectdb.html#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID =</span> dict_CO[[<span class="st">&quot;ID&quot;</span>]], <span class="at">Description =</span> dict_CO[[<span class="st">&quot;Description&quot;</span>]],</span>
<span id="cb17-3"><a href="collectdb.html#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">Count =</span> <span class="cn">NA</span>, <span class="at">Gene =</span> <span class="cn">NA</span>, <span class="at">GeneID =</span> <span class="cn">NA</span></span>
<span id="cb17-4"><a href="collectdb.html#cb17-4" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb17-5"><a href="collectdb.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(map)){</span>
<span id="cb17-6"><a href="collectdb.html#cb17-6" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(dict_CO[<span class="fu">which</span>(dict_CO[[<span class="st">&quot;ID&quot;</span>]] <span class="sc">==</span> map[[<span class="st">&quot;ID&quot;</span>]][i]),]<span class="sc">$</span>Symbol)</span>
<span id="cb17-7"><a href="collectdb.html#cb17-7" aria-hidden="true" tabindex="-1"></a>  map<span class="sc">$</span>GeneID[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>(genes, <span class="at">collapse =</span> <span class="st">&quot;/&quot;</span>)</span>
<span id="cb17-8"><a href="collectdb.html#cb17-8" aria-hidden="true" tabindex="-1"></a>  map<span class="sc">$</span>Count[i] <span class="ot">&lt;-</span> <span class="fu">length</span>(genes)</span>
<span id="cb17-9"><a href="collectdb.html#cb17-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-10"><a href="collectdb.html#cb17-10" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(tidy_DO[[<span class="st">&quot;disease&quot;</span>]], map)</span>
<span id="cb17-11"><a href="collectdb.html#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="collectdb.html#cb17-12" aria-hidden="true" tabindex="-1"></a>tidy_CB <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb17-13"><a href="collectdb.html#cb17-13" aria-hidden="true" tabindex="-1"></a>tidy_CB[[<span class="st">&quot;cell&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">rbind</span>(tmp, res)</span>
<span id="cb17-14"><a href="collectdb.html#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tidy_CB[[<span class="st">&quot;cell&quot;</span>]]) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tidy_CB[[<span class="st">&quot;cell&quot;</span>]])</span></code></pre></div>
<p>Save the result.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="collectdb.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tidy_CB, <span class="at">file =</span> <span class="st">&quot;data/2020_001_databases/20210817_tidy_CB_human.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="markergenes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cell_type.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to ASURAT (ver. 0.0.0.9000).pdf", "Introduction to ASURAT (ver. 0.0.0.9000).epub"],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
