<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Introduction to ASURAT (version 0.0.0.9000)</title>
  <meta name="description" content="This is a developmental version of ASURAT." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Introduction to ASURAT (version 0.0.0.9000)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a developmental version of ASURAT." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to ASURAT (version 0.0.0.9000)" />
  
  <meta name="twitter:description" content="This is a developmental version of ASURAT." />
  

<meta name="author" content="Keita Iida" />


<meta name="date" content="2021-07-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="dataset.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#license-and-description"><i class="fa fa-check"></i><b>1.1</b> License and description</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computational-environment"><i class="fa fa-check"></i><b>1.2</b> Computational environment</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dataset.html"><a href="dataset.html"><i class="fa fa-check"></i><b>2</b> Preparing data sets</a><ul>
<li class="chapter" data-level="2.1" data-path="dataset.html"><a href="dataset.html#case-i-input-gene-expression-matrix-with-row-and-column-names"><i class="fa fa-check"></i><b>2.1</b> Case I: input gene expression matrix with row and column names</a></li>
<li class="chapter" data-level="2.2" data-path="dataset.html"><a href="dataset.html#case-ii-input-10x-data"><i class="fa fa-check"></i><b>2.2</b> Case II: input 10x data</a></li>
<li class="chapter" data-level="2.3" data-path="dataset.html"><a href="dataset.html#case-iii-input-10x-pbmc-data"><i class="fa fa-check"></i><b>2.3</b> Case III: input 10x PBMC data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataqc.html"><a href="dataqc.html"><i class="fa fa-check"></i><b>3</b> Data quality control (QC)</a><ul>
<li class="chapter" data-level="3.1" data-path="dataqc.html"><a href="dataqc.html#quick-qc"><i class="fa fa-check"></i><b>3.1</b> Quick QC</a></li>
<li class="chapter" data-level="3.2" data-path="dataqc.html"><a href="dataqc.html#sample-qc"><i class="fa fa-check"></i><b>3.2</b> Sample QC</a></li>
<li class="chapter" data-level="3.3" data-path="dataqc.html"><a href="dataqc.html#variable-qc"><i class="fa fa-check"></i><b>3.3</b> Variable QC</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>4</b> Normalizing and centering data</a></li>
<li class="chapter" data-level="5" data-path="computcorr.html"><a href="computcorr.html"><i class="fa fa-check"></i><b>5</b> Computing correlations among genes</a></li>
<li class="chapter" data-level="6" data-path="markergenes.html"><a href="markergenes.html"><i class="fa fa-check"></i><b>6</b> Checking expression profiles of marker genes</a></li>
<li class="chapter" data-level="7" data-path="collectdb.html"><a href="collectdb.html"><i class="fa fa-check"></i><b>7</b> Collecting databases (optional)</a><ul>
<li class="chapter" data-level="7.1" data-path="collectdb.html"><a href="collectdb.html#disease-ontology-db"><i class="fa fa-check"></i><b>7.1</b> Disease Ontology DB</a></li>
<li class="chapter" data-level="7.2" data-path="collectdb.html"><a href="collectdb.html#cell-ontology-db"><i class="fa fa-check"></i><b>7.2</b> Cell Ontology DB</a></li>
<li class="chapter" data-level="7.3" data-path="collectdb.html"><a href="collectdb.html#gene-ontology-db"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology DB</a></li>
<li class="chapter" data-level="7.4" data-path="collectdb.html"><a href="collectdb.html#kegg"><i class="fa fa-check"></i><b>7.4</b> KEGG</a></li>
<li class="chapter" data-level="7.5" data-path="collectdb.html"><a href="collectdb.html#reactome"><i class="fa fa-check"></i><b>7.5</b> Reactome</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html"><i class="fa fa-check"></i><b>8</b> ASURAT using Disease Ontology database</a><ul>
<li class="chapter" data-level="8.1" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#creating-sign-by-sample-matrices"><i class="fa fa-check"></i><b>8.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="8.2" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="8.3" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#cell-clustering"><i class="fa fa-check"></i><b>8.3</b> Cell clustering</a></li>
<li class="chapter" data-level="8.4" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#visualization"><i class="fa fa-check"></i><b>8.4</b> Visualization</a></li>
<li class="chapter" data-level="8.5" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#significant-signs"><i class="fa fa-check"></i><b>8.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html"><i class="fa fa-check"></i><b>9</b> ASURAT using Cell Ontology database (optional)</a><ul>
<li class="chapter" data-level="9.1" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#creating-sign-by-sample-matrices-1"><i class="fa fa-check"></i><b>9.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="9.2" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>9.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="9.3" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#cell-clustering-1"><i class="fa fa-check"></i><b>9.3</b> Cell clustering</a></li>
<li class="chapter" data-level="9.4" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#visualization-1"><i class="fa fa-check"></i><b>9.4</b> Visualization</a></li>
<li class="chapter" data-level="9.5" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#significant-signs-1"><i class="fa fa-check"></i><b>9.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html"><i class="fa fa-check"></i><b>10</b> ASURAT using Gene Ontology database (optional)</a><ul>
<li class="chapter" data-level="10.1" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#creating-sign-by-sample-matrices-2"><i class="fa fa-check"></i><b>10.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="10.2" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>10.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="10.3" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#cell-clustering-2"><i class="fa fa-check"></i><b>10.3</b> Cell clustering</a></li>
<li class="chapter" data-level="10.4" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#visualization-2"><i class="fa fa-check"></i><b>10.4</b> Visualization</a></li>
<li class="chapter" data-level="10.5" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#significant-signs-2"><i class="fa fa-check"></i><b>10.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html"><i class="fa fa-check"></i><b>11</b> ASURAT using KEGG (optional)</a><ul>
<li class="chapter" data-level="11.1" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#creating-sign-by-sample-matrices-3"><i class="fa fa-check"></i><b>11.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="11.2" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#dimensionality-reduction-3"><i class="fa fa-check"></i><b>11.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="11.3" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#cell-clustering-3"><i class="fa fa-check"></i><b>11.3</b> Cell clustering</a></li>
<li class="chapter" data-level="11.4" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#visualizing"><i class="fa fa-check"></i><b>11.4</b> Visualizing</a></li>
<li class="chapter" data-level="11.5" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#significant-signs-3"><i class="fa fa-check"></i><b>11.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html"><i class="fa fa-check"></i><b>12</b> ASURAT using Reactome (optional)</a><ul>
<li class="chapter" data-level="12.1" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#creating-sign-by-sample-matrices-4"><i class="fa fa-check"></i><b>12.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="12.2" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#dimensionality-reduction-4"><i class="fa fa-check"></i><b>12.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="12.3" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#sample-clustering"><i class="fa fa-check"></i><b>12.3</b> Sample clustering</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-sign-analysis-by-concatenating-do-co-go-kegg-and-reactome.html"><a href="multiple-sign-analysis-by-concatenating-do-co-go-kegg-and-reactome.html"><i class="fa fa-check"></i><b>13</b> Multiple sign analysis by concatenating DO, CO, GO, KEGG, and Reactome</a></li>
<li class="chapter" data-level="14" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><i class="fa fa-check"></i><b>14</b> Appendix A: comparing performances of ASURAT and existing methods</a><ul>
<li class="chapter" data-level="14.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#installation-1"><i class="fa fa-check"></i><b>14.1</b> Installation</a></li>
<li class="chapter" data-level="14.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#removing-the-outlier-cells-detected-by-asurat"><i class="fa fa-check"></i><b>14.2</b> Removing the outlier cells detected by ASURAT</a></li>
<li class="chapter" data-level="14.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#assessing-clustering-performances"><i class="fa fa-check"></i><b>14.3</b> Assessing clustering performances</a><ul>
<li class="chapter" data-level="14.3.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#seurat"><i class="fa fa-check"></i><b>14.3.1</b> Seurat</a></li>
<li class="chapter" data-level="14.3.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#monocle-3"><i class="fa fa-check"></i><b>14.3.2</b> Monocle 3</a></li>
<li class="chapter" data-level="14.3.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#sc3"><i class="fa fa-check"></i><b>14.3.3</b> SC3</a></li>
<li class="chapter" data-level="14.3.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>14.3.4</b> Principal component analysis (PCA)</a></li>
<li class="chapter" data-level="14.3.5" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#asurat"><i class="fa fa-check"></i><b>14.3.5</b> ASURAT</a></li>
<li class="chapter" data-level="14.3.6" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#comparing-clustering-validity-indices"><i class="fa fa-check"></i><b>14.3.6</b> Comparing clustering validity indices</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#cell-type-inference-in-pbmc_4000-and-pbmc_6000"><i class="fa fa-check"></i><b>14.4</b> Cell-type inference in <code>pbmc_4000</code> and <code>pbmc_6000</code></a><ul>
<li class="chapter" data-level="14.4.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#seurat-1"><i class="fa fa-check"></i><b>14.4.1</b> Seurat</a></li>
<li class="chapter" data-level="14.4.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#monocle-3-1"><i class="fa fa-check"></i><b>14.4.2</b> Monocle 3</a></li>
<li class="chapter" data-level="14.4.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#sc3-1"><i class="fa fa-check"></i><b>14.4.3</b> SC3</a></li>
<li class="chapter" data-level="14.4.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#asurat-1"><i class="fa fa-check"></i><b>14.4.4</b> ASURAT</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#identifying-cell-types-in-pbmc_4000-and-pbmc_6000"><i class="fa fa-check"></i><b>14.5</b> Identifying cell types in <code>pbmc_4000</code> and <code>pbmc_6000</code></a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="appendix-b-automatically-tuning-asurats-parameters.html"><a href="appendix-b-automatically-tuning-asurats-parameters.html"><i class="fa fa-check"></i><b>15</b> Appendix B: automatically tuning ASURATâ€™s parameters</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to ASURAT (version 0.0.0.9000)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Introduction to ASURAT (version 0.0.0.9000)</h1>
<p class="author"><em>Keita Iida</em></p>
<p class="date"><em>2021-07-01</em></p>
</div>
<div id="overview" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Overview</h1>
<p>ASURAT is a single-cell RNA sequencing (scRNA-seq) data analysis pipeline,
developed for simultaneously clustering single-cell transcriptomes
and biological interpretation.
Inputting an scRNA-seq data and knowledge-based databases (DBs), ASURAT creates
lists of biological terms with respect to cell type and biological functions.
Then, ASURAT creates a functional spectrum matrix.
By analyzing this matrix, users can cluster cells to aid their
interpretation.</p>
<div id="license-and-description" class="section level2">
<h2><span class="header-section-number">1.1</span> License and description</h2>
<pre><code>Package: ASURAT
Title: Functional annotation-driven unsupervised clustering of scRNA-seq data
Version: 0.0.0.9000
Authors@R: person(&quot;Keita&quot;, &quot;Iida&quot;, email = &quot;kiida@protein.osaka-u.ac.jp&quot;,
  role = c(&quot;aut&quot;, &quot;cre&quot;))
Description: In single-cell transcriptomics ASURAT assists users to cluster
    cells in biologically explainable manners and find various types of
    biological terms specifically upregulated in each cluster.
    Using ASURAT, users can create and analyze a special type of matrix termed
    &quot;sign-by-sample matrix&quot;, of which the rows and columns stand for
    biological terms and samples, respectively.
License: GPL-3
Encoding: UTF-8
LazyData: true
Imports: Rcpp (&gt;= 1.0.6)
LinkingTo: Rcpp
Suggests: 
    rmarkdown,
    knitr,
    Seurat (&gt;= 3.0.0)
RoxygenNote: 7.1.1
Depends: 
    R (&gt;= 2.10)
VignetteBuilder: knitr</code></pre>
</div>
<div id="computational-environment" class="section level2">
<h2><span class="header-section-number">1.2</span> Computational environment</h2>
<p>MacBook Pro (Big Sur, 16-inch, 2019), Processor (2.4 GHz 8-Core Intel Core i9),
Memory (64 GB 2667 MHz DDR4).</p>
</div>
<div id="installation" class="section level2">
<h2><span class="header-section-number">1.3</span> Installation</h2>
<p>The following libraries are used almost everywhere.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1"></a><span class="co"># General</span></span>
<span id="cb2-2"><a href="index.html#cb2-2"></a><span class="kw">library</span>(tidyverse)       <span class="co"># For efficient handling of data.frame</span></span>
<span id="cb2-3"><a href="index.html#cb2-3"></a><span class="kw">library</span>(Matrix)          <span class="co"># For using `readMM`</span></span>
<span id="cb2-4"><a href="index.html#cb2-4"></a></span>
<span id="cb2-5"><a href="index.html#cb2-5"></a><span class="co"># Below is a memo for attaching org.Hs.eg.db</span></span>
<span id="cb2-6"><a href="index.html#cb2-6"></a><span class="co"># options(connectionObserver = NULL)</span></span>
<span id="cb2-7"><a href="index.html#cb2-7"></a><span class="kw">library</span>(org.Hs.eg.db)    <span class="co"># For using human genome annotation package</span></span>
<span id="cb2-8"><a href="index.html#cb2-8"></a></span>
<span id="cb2-9"><a href="index.html#cb2-9"></a><span class="co"># Plot</span></span>
<span id="cb2-10"><a href="index.html#cb2-10"></a><span class="kw">library</span>(ggExtra)         <span class="co"># For graphics</span></span>
<span id="cb2-11"><a href="index.html#cb2-11"></a><span class="kw">library</span>(ggrepel)         <span class="co"># For graphics with labeling</span></span>
<span id="cb2-12"><a href="index.html#cb2-12"></a><span class="kw">library</span>(plot3D)          <span class="co"># For 3-dimensional scatter plot</span></span>
<span id="cb2-13"><a href="index.html#cb2-13"></a><span class="kw">library</span>(ComplexHeatmap)  <span class="co"># For ComplexHeatmap</span></span>
<span id="cb2-14"><a href="index.html#cb2-14"></a><span class="kw">library</span>(circlize)        <span class="co"># For color mapping function in ComplexHeatmap</span></span>
<span id="cb2-15"><a href="index.html#cb2-15"></a><span class="kw">library</span>(DT)              <span class="co"># For using datatable</span></span>
<span id="cb2-16"><a href="index.html#cb2-16"></a><span class="kw">library</span>(viridis)         <span class="co"># For using color palette `plasma`</span></span>
<span id="cb2-17"><a href="index.html#cb2-17"></a></span>
<span id="cb2-18"><a href="index.html#cb2-18"></a><span class="co"># Normalization</span></span>
<span id="cb2-19"><a href="index.html#cb2-19"></a><span class="kw">library</span>(bayNorm)         <span class="co"># For using bayNorm</span></span>
<span id="cb2-20"><a href="index.html#cb2-20"></a><span class="co"># Below is a memo for the install of bayNorm:</span></span>
<span id="cb2-21"><a href="index.html#cb2-21"></a><span class="co"># if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb2-22"><a href="index.html#cb2-22"></a><span class="co">#     install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb2-23"><a href="index.html#cb2-23"></a><span class="co"># BiocManager::install(c(&quot;bayNorm&quot;))</span></span>
<span id="cb2-24"><a href="index.html#cb2-24"></a></span>
<span id="cb2-25"><a href="index.html#cb2-25"></a><span class="co"># Utilities</span></span>
<span id="cb2-26"><a href="index.html#cb2-26"></a><span class="kw">library</span>(Rtsne)           <span class="co"># For using Rtsne</span></span>
<span id="cb2-27"><a href="index.html#cb2-27"></a><span class="kw">library</span>(umap)            <span class="co"># For using umap</span></span>
<span id="cb2-28"><a href="index.html#cb2-28"></a><span class="kw">library</span>(Rcpp)            <span class="co"># For using external C++ codes</span></span>
<span id="cb2-29"><a href="index.html#cb2-29"></a><span class="kw">library</span>(cluster)         <span class="co"># For using pam</span></span>
<span id="cb2-30"><a href="index.html#cb2-30"></a><span class="kw">library</span>(Seurat)          <span class="co"># For using Seurat</span></span>
<span id="cb2-31"><a href="index.html#cb2-31"></a><span class="kw">library</span>(destiny)         <span class="co"># For using diffusion map</span></span>
<span id="cb2-32"><a href="index.html#cb2-32"></a><span class="kw">library</span>(merlot)          <span class="co"># For detecting branches along pseudotime</span></span>
<span id="cb2-33"><a href="index.html#cb2-33"></a><span class="co"># Below is a memo for the install of merlot:</span></span>
<span id="cb2-34"><a href="index.html#cb2-34"></a><span class="co"># remotes::install_github(&quot;soedinglab/merlot&quot;)</span></span>
<span id="cb2-35"><a href="index.html#cb2-35"></a></span>
<span id="cb2-36"><a href="index.html#cb2-36"></a><span class="co"># Databases</span></span>
<span id="cb2-37"><a href="index.html#cb2-37"></a><span class="kw">library</span>(DOSE)            <span class="co"># For using DB for Disease Ontology</span></span>
<span id="cb2-38"><a href="index.html#cb2-38"></a><span class="kw">library</span>(ontoProc)        <span class="co"># For using DB for Cell Ontology</span></span>
<span id="cb2-39"><a href="index.html#cb2-39"></a><span class="kw">library</span>(clusterProfiler) <span class="co"># For using DB for Gene Ontology</span></span>
<span id="cb2-40"><a href="index.html#cb2-40"></a><span class="kw">library</span>(GOSemSim)        <span class="co"># For using godata in GOSemSim package</span></span>
<span id="cb2-41"><a href="index.html#cb2-41"></a><span class="kw">library</span>(KEGGREST)        <span class="co"># For using KEGG</span></span>
<span id="cb2-42"><a href="index.html#cb2-42"></a><span class="kw">library</span>(reactome.db)     <span class="co"># For using Reactome</span></span></code></pre></div>
<p>Check the R version, which is recommended to be 4.0 or later.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1"></a>version</span>
<span id="cb3-2"><a href="index.html#cb3-2"></a><span class="co">#&gt;                _                           </span></span>
<span id="cb3-3"><a href="index.html#cb3-3"></a><span class="co">#&gt; platform       x86_64-apple-darwin17.0     </span></span>
<span id="cb3-4"><a href="index.html#cb3-4"></a><span class="co">#&gt; arch           x86_64                      </span></span>
<span id="cb3-5"><a href="index.html#cb3-5"></a><span class="co">#&gt; os             darwin17.0                  </span></span>
<span id="cb3-6"><a href="index.html#cb3-6"></a><span class="co">#&gt; system         x86_64, darwin17.0          </span></span>
<span id="cb3-7"><a href="index.html#cb3-7"></a><span class="co">#&gt; status                                     </span></span>
<span id="cb3-8"><a href="index.html#cb3-8"></a><span class="co">#&gt; major          4                           </span></span>
<span id="cb3-9"><a href="index.html#cb3-9"></a><span class="co">#&gt; minor          0.4                         </span></span>
<span id="cb3-10"><a href="index.html#cb3-10"></a><span class="co">#&gt; year           2021                        </span></span>
<span id="cb3-11"><a href="index.html#cb3-11"></a><span class="co">#&gt; month          02                          </span></span>
<span id="cb3-12"><a href="index.html#cb3-12"></a><span class="co">#&gt; day            15                          </span></span>
<span id="cb3-13"><a href="index.html#cb3-13"></a><span class="co">#&gt; svn rev        80002                       </span></span>
<span id="cb3-14"><a href="index.html#cb3-14"></a><span class="co">#&gt; language       R                           </span></span>
<span id="cb3-15"><a href="index.html#cb3-15"></a><span class="co">#&gt; version.string R version 4.0.4 (2021-02-15)</span></span>
<span id="cb3-16"><a href="index.html#cb3-16"></a><span class="co">#&gt; nickname       Lost Library Book</span></span></code></pre></div>
<pre><code>platform       x86_64-apple-darwin17.0     
arch           x86_64                      
os             darwin17.0                  
system         x86_64, darwin17.0          
status                                     
major          4                           
minor          0.4                         
year           2021                        
month          02                          
day            15                          
svn rev        80002                       
language       R                           
version.string R version 4.0.4 (2021-02-15)
nickname       Lost Library Book           </code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="index.html#cb5-1"></a><span class="kw">sessionInfo</span>()</span>
<span id="cb5-2"><a href="index.html#cb5-2"></a><span class="co">#&gt; R version 4.0.4 (2021-02-15)</span></span>
<span id="cb5-3"><a href="index.html#cb5-3"></a><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span id="cb5-4"><a href="index.html#cb5-4"></a><span class="co">#&gt; Running under: macOS Big Sur 10.16</span></span>
<span id="cb5-5"><a href="index.html#cb5-5"></a><span class="co">#&gt; </span></span>
<span id="cb5-6"><a href="index.html#cb5-6"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb5-7"><a href="index.html#cb5-7"></a><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span></span>
<span id="cb5-8"><a href="index.html#cb5-8"></a><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span></span>
<span id="cb5-9"><a href="index.html#cb5-9"></a><span class="co">#&gt; </span></span>
<span id="cb5-10"><a href="index.html#cb5-10"></a><span class="co">#&gt; locale:</span></span>
<span id="cb5-11"><a href="index.html#cb5-11"></a><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span id="cb5-12"><a href="index.html#cb5-12"></a><span class="co">#&gt; </span></span>
<span id="cb5-13"><a href="index.html#cb5-13"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb5-14"><a href="index.html#cb5-14"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb5-15"><a href="index.html#cb5-15"></a><span class="co">#&gt; </span></span>
<span id="cb5-16"><a href="index.html#cb5-16"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb5-17"><a href="index.html#cb5-17"></a><span class="co">#&gt;  [1] bookdown_0.21     digest_0.6.27     R6_2.5.0          jsonlite_1.7.2   </span></span>
<span id="cb5-18"><a href="index.html#cb5-18"></a><span class="co">#&gt;  [5] magrittr_2.0.1    evaluate_0.14     stringi_1.5.3     rlang_0.4.10     </span></span>
<span id="cb5-19"><a href="index.html#cb5-19"></a><span class="co">#&gt;  [9] rstudioapi_0.13   jquerylib_0.1.3   bslib_0.2.4       rmarkdown_2.7    </span></span>
<span id="cb5-20"><a href="index.html#cb5-20"></a><span class="co">#&gt; [13] tools_4.0.4       stringr_1.4.0     xfun_0.22         yaml_2.2.1       </span></span>
<span id="cb5-21"><a href="index.html#cb5-21"></a><span class="co">#&gt; [17] compiler_4.0.4    htmltools_0.5.1.1 knitr_1.32        sass_0.3.1</span></span></code></pre></div>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3          plotly_4.9.3           reactome.db_1.74.0     KEGGREST_1.30.1       
 [5] GOSemSim_2.16.1        clusterProfiler_3.18.1 ontoProc_1.12.0        ontologyIndex_2.7     
 [9] DOSE_3.16.0            merlot_0.2.2           destiny_3.4.0          SeuratObject_4.0.0    
[13] Seurat_4.0.1           cluster_2.1.2          Rcpp_1.0.6             umap_0.2.7.0          
[17] Rtsne_0.15             limma_3.46.0           sva_3.38.0             BiocParallel_1.24.1   
[21] genefilter_1.72.1      mgcv_1.8-35            nlme_3.1-152           bayNorm_1.8.0         
[25] viridis_0.6.0          viridisLite_0.4.0      DT_0.18                circlize_0.4.12       
[29] ComplexHeatmap_2.6.2   plot3D_1.3             ggrepel_0.9.1          ggExtra_0.9           
[33] org.Hs.eg.db_3.12.0    AnnotationDbi_1.52.0   IRanges_2.24.1         S4Vectors_0.28.1      
[37] Biobase_2.50.0         BiocGenerics_0.36.0    Matrix_1.3-2           forcats_0.5.1         
[41] stringr_1.4.0          dplyr_1.0.5            purrr_0.3.4            readr_1.4.0           
[45] tidyr_1.1.3            tibble_3.1.1           ggplot2_3.3.3          tidyverse_1.3.1       

loaded via a namespace (and not attached):
  [1] rgl_0.106.6                 vcd_1.4-8                   ica_1.0-2                  
  [4] class_7.3-18                foreach_1.5.1               lmtest_0.9-38              
  [7] crayon_1.4.1                laeken_0.5.1                spatstat.core_2.1-2        
 [10] MASS_7.3-53.1               backports_1.2.1             reprex_2.0.0               
 [13] rlang_0.4.10                XVector_0.30.0              ROCR_1.0-11                
 [16] readxl_1.3.1                irlba_2.3.3                 smoother_1.1               
 [19] rjson_0.2.20                manipulateWidget_0.10.1     bit64_4.0.5                
 [22] glue_1.4.2                  sctransform_0.3.2           spatstat.sparse_2.0-0      
 [25] dotCall64_1.0-1             tcltk_4.0.4                 spatstat.geom_2.1-0        
 [28] haven_2.4.0                 tidyselect_1.1.0            SummarizedExperiment_1.20.0
 [31] rio_0.5.26                  fitdistrplus_1.1-3          XML_3.99-0.6               
 [34] zoo_1.8-9                   xtable_1.8-4                RcppHNSW_0.3.0             
 [37] magrittr_2.0.1              evaluate_0.14               cli_2.4.0                  
 [40] zlibbioc_1.36.0             rstudioapi_0.13             miniUI_0.1.1.1             
 [43] sp_1.4-5                    bslib_0.2.4                 rpart_4.1-15               
 [46] fastmatch_1.1-0             RcppEigen_0.3.3.9.1         maps_3.3.0                 
 [49] fields_11.6                 shiny_1.6.0                 xfun_0.22                  
 [52] askpass_1.1                 clue_0.3-59                 tidygraph_1.2.0            
 [55] pcaMethods_1.82.0           doSNOW_1.0.19               ElPiGraph.R_1.0.0          
 [58] listenv_0.8.0               Biostrings_2.58.0           png_0.1-7                  
 [61] future_1.21.0               withr_2.4.2                 bitops_1.0-6               
 [64] ggforce_0.3.3               ranger_0.12.1               plyr_1.8.6                 
 [67] cellranger_1.1.0            e1071_1.7-6                 pillar_1.6.0               
 [70] GlobalOptions_0.1.2         cachem_1.0.4                fs_1.5.0                   
 [73] scatterplot3d_0.3-41        TTR_0.24.2                  GetoptLong_1.0.5           
 [76] xts_0.12.1                  vctrs_0.3.7                 ellipsis_0.3.1             
 [79] generics_0.1.0              tools_4.0.4                 foreign_0.8-81             
 [82] munsell_0.5.0               tweenr_1.0.2                fgsea_1.16.0               
 [85] proxy_0.4-25                DelayedArray_0.16.3         fastmap_1.1.0              
 [88] compiler_4.0.4              paintmap_1.0                abind_1.4-5                
 [91] httpuv_1.5.5                GenomeInfoDbData_1.2.4      edgeR_3.32.1               
 [94] lattice_0.20-41             deldir_0.2-10               snow_0.4-3                 
 [97] utf8_1.2.1                  later_1.1.0.1               BiocFileCache_1.14.0       
[100] jsonlite_1.7.2              ggplot.multistats_1.0.0     scales_1.1.1               
[103] graph_1.68.0                pbapply_1.4-3               carData_3.0-4              
[106] lazyeval_0.2.2              promises_1.2.0.1            car_3.0-10                 
[109] goftest_1.2-2               spatstat.utils_2.1-0        reticulate_1.19            
[112] rmarkdown_2.7               openxlsx_4.2.3              cowplot_1.1.1              
[115] webshot_0.5.2               downloader_0.4              uwot_0.1.10                
[118] igraph_1.2.6                survival_3.2-10             yaml_2.2.1                 
[121] htmltools_0.5.1.1           memoise_2.0.0               locfit_1.5-9.4             
[124] graphlayouts_0.7.1          digest_0.6.27               assertthat_0.2.1           
[127] mime_0.10                   rappdirs_0.3.3              spam_2.6-0                 
[130] RSQLite_2.2.6               LSD_4.1-0                   future.apply_1.7.0         
[133] misc3d_0.9-0                data.table_1.14.0           blob_1.2.1                 
[136] labeling_0.4.2              splines_4.0.4               Cairo_1.5-12.2             
[139] RCurl_1.98-1.3              broom_0.7.6                 hms_1.0.0                  
[142] modelr_0.1.8                colorspace_2.0-0            BiocManager_1.30.12        
[145] GenomicRanges_1.42.0        shape_1.4.5                 nnet_7.3-15                
[148] sass_0.3.1                  bookdown_0.21               RANN_2.6.1                 
[151] enrichplot_1.10.2           fansi_0.4.2                 VIM_6.1.0                  
[154] parallelly_1.24.0           R6_2.5.0                    ggridges_0.5.3             
[157] lifecycle_1.0.0             zip_2.1.1                   curl_4.3                   
[160] jquerylib_0.1.3             leiden_0.3.7                robustbase_0.93-7          
[163] DO.db_2.9                   qvalue_2.22.0               RcppAnnoy_0.0.18           
[166] RColorBrewer_1.1-2          iterators_1.0.13            htmlwidgets_1.5.3          
[169] polyclip_1.10-0             crosstalk_1.1.1             shadowtext_0.0.7           
[172] rvest_1.0.0                 globals_0.14.0              openssl_1.4.3              
[175] patchwork_1.1.1             codetools_0.2-18            matrixStats_0.58.0         
[178] lubridate_1.7.10            GO.db_3.12.1                SingleCellExperiment_1.12.0
[181] dbplyr_2.1.1                RSpectra_0.16-0             GenomeInfoDb_1.26.7        
[184] gtable_0.3.0                DBI_1.1.1                   tensor_1.5                 
[187] httr_1.4.2                  KernSmooth_2.23-18          stringi_1.5.3              
[190] reshape2_1.4.4              farver_2.1.0                ontologyPlot_1.6           
[193] annotate_1.68.0             ggthemes_4.2.4              hexbin_1.28.2              
[196] Rgraphviz_2.34.0            xml2_1.3.2                  rvcheck_0.1.8              
[199] boot_1.3-27                 scattermore_0.7             DEoptimR_1.0-8             
[202] bit_4.0.4                   scatterpie_0.1.5            MatrixGenerics_1.2.1       
[205] spatstat.data_2.1-0         ggraph_2.0.5                pkgconfig_2.0.3            
[208] knn.covertree_1.0           knitr_1.32</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="dataset.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to ASURAT (ver. 0.0.0.9000).pdf", "Introduction to ASURAT (ver. 0.0.0.9000).epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
